<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Cordeliers Artificial Intelligence and Bioinformatics">

<title>README for params.yml Configuration File</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#structure-of-params.yml" id="toc-structure-of-params.yml" class="nav-link active" data-scroll-target="#structure-of-params.yml">Structure of <code>params.yml</code></a>
  <ul class="collapse">
  <li><a href="#project-information" id="toc-project-information" class="nav-link" data-scroll-target="#project-information">Project Information</a></li>
  <li><a href="#species-information" id="toc-species-information" class="nav-link" data-scroll-target="#species-information">Species Information</a></li>
  <li><a href="#plot-annotations" id="toc-plot-annotations" class="nav-link" data-scroll-target="#plot-annotations">Plot Annotations</a></li>
  <li><a href="#quality-control-qc" id="toc-quality-control-qc" class="nav-link" data-scroll-target="#quality-control-qc">Quality Control (QC)</a></li>
  <li><a href="#clustering-configuration" id="toc-clustering-configuration" class="nav-link" data-scroll-target="#clustering-configuration">Clustering Configuration</a></li>
  <li><a href="#pathway-collections" id="toc-pathway-collections" class="nav-link" data-scroll-target="#pathway-collections">Pathway Collections</a></li>
  <li><a href="#targeted-visualizations" id="toc-targeted-visualizations" class="nav-link" data-scroll-target="#targeted-visualizations">Targeted Visualizations</a></li>
  </ul></li>
  <li><a href="#example-params.yml-file" id="toc-example-params.yml-file" class="nav-link" data-scroll-target="#example-params.yml-file">Example <code>params.yml</code> File</a></li>
  <li><a href="#editing-tips" id="toc-editing-tips" class="nav-link" data-scroll-target="#editing-tips">Editing Tips</a></li>
  <li><a href="#common-issues" id="toc-common-issues" class="nav-link" data-scroll-target="#common-issues">Common Issues</a></li>
  <li><a href="#structure-of-params.yml-1" id="toc-structure-of-params.yml-1" class="nav-link" data-scroll-target="#structure-of-params.yml-1">Structure of <code>params.yml</code></a>
  <ul class="collapse">
  <li><a href="#project-information-1" id="toc-project-information-1" class="nav-link" data-scroll-target="#project-information-1">Project Information</a></li>
  <li><a href="#species-information-1" id="toc-species-information-1" class="nav-link" data-scroll-target="#species-information-1">Species Information</a></li>
  <li><a href="#plot-annotations-1" id="toc-plot-annotations-1" class="nav-link" data-scroll-target="#plot-annotations-1">Plot Annotations</a></li>
  <li><a href="#quality-control-qc-1" id="toc-quality-control-qc-1" class="nav-link" data-scroll-target="#quality-control-qc-1">Quality Control (QC)</a></li>
  <li><a href="#clustering-configuration-1" id="toc-clustering-configuration-1" class="nav-link" data-scroll-target="#clustering-configuration-1">Clustering Configuration</a></li>
  <li><a href="#pathway-collections-1" id="toc-pathway-collections-1" class="nav-link" data-scroll-target="#pathway-collections-1">Pathway Collections</a></li>
  <li><a href="#targeted-visualizations-1" id="toc-targeted-visualizations-1" class="nav-link" data-scroll-target="#targeted-visualizations-1">Targeted Visualizations</a></li>
  </ul></li>
  <li><a href="#example-params.yml-file-1" id="toc-example-params.yml-file-1" class="nav-link" data-scroll-target="#example-params.yml-file-1">Example <code>params.yml</code> File</a></li>
  <li><a href="#editing-tips-1" id="toc-editing-tips-1" class="nav-link" data-scroll-target="#editing-tips-1">Editing Tips</a></li>
  <li><a href="#common-issues-1" id="toc-common-issues-1" class="nav-link" data-scroll-target="#common-issues-1">Common Issues</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">README for <code>params.yml</code> Configuration File</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Cordeliers Artificial Intelligence and Bioinformatics </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<p>This <code>params.yml</code> file is used to configure the unsupervised RNA-seq analysis pipeline. It allows users to specify settings for clustering, filtering, normalization, and visualization of their RNA-seq dataset. Below, each parameter is explained in detail to help guide you in filling out or understanding this configuration.</p>
<section id="structure-of-params.yml" class="level2">
<h2 class="anchored" data-anchor-id="structure-of-params.yml">Structure of <code>params.yml</code></h2>
<p>The file is formatted in YAML, which makes it easy to read and edit. Below, we break down the purpose of each parameter.</p>
<section id="project-information" class="level3">
<h3 class="anchored" data-anchor-id="project-information">Project Information</h3>
<section id="project" class="level4">
<h4 class="anchored" data-anchor-id="project"><code>project</code></h4>
<ul>
<li><strong>Description</strong>: A descriptive name for the project.</li>
<li><strong>Example</strong>: <code>Mouse STING KO</code></li>
<li><strong>Usage</strong>: This will be used as the title of the analysis report to identify the project.</li>
</ul>
</section>
<section id="data_file" class="level4">
<h4 class="anchored" data-anchor-id="data_file"><code>data_file</code></h4>
<ul>
<li><strong>Description</strong>: Path to the RNA-seq data file in <code>.RDS</code> format.</li>
<li><strong>Example</strong>: <code>data/STING_KO/exp_data.RDS</code></li>
<li><strong>Usage</strong>: This should point to the expression data file that you wish to analyze. The <code>.RDS</code> file contains your pre-processed gene expression matrix.</li>
</ul>
</section>
<section id="use_symbol" class="level4">
<h4 class="anchored" data-anchor-id="use_symbol"><code>use_symbol</code></h4>
<ul>
<li><strong>Description</strong>: Whether to convert gene identifiers to gene symbols.</li>
<li><strong>Options</strong>: <code>true</code> or <code>false</code></li>
<li><strong>Example</strong>: <code>true</code></li>
<li><strong>Usage</strong>: Set to <code>true</code> to rebase Ensembl IDs to gene symbols, which can make downstream analysis and interpretation easier for biological studies.</li>
</ul>
</section>
<section id="show_code" class="level4">
<h4 class="anchored" data-anchor-id="show_code"><code>show_code</code></h4>
<ul>
<li><strong>Description</strong>: Whether to display the code in the output report.</li>
<li><strong>Options</strong>: <code>true</code> or <code>false</code></li>
<li><strong>Example</strong>: <code>false</code></li>
<li><strong>Usage</strong>: Set this to <code>true</code> if you want to include R code in the final output report, which may help for reproducibility or educational purposes.</li>
</ul>
</section>
</section>
<section id="species-information" class="level3">
<h3 class="anchored" data-anchor-id="species-information">Species Information</h3>
<section id="species" class="level4">
<h4 class="anchored" data-anchor-id="species"><code>species</code></h4>
<ul>
<li><strong>Description</strong>: Specifies the species of the dataset.</li>
<li><strong>Example</strong>: <code>Mus musculus</code></li>
<li><strong>Usage</strong>: This parameter helps select the appropriate gene sets and pathways for the analysis. Here, <code>Mus musculus</code> (mouse) is specified.</li>
</ul>
</section>
</section>
<section id="plot-annotations" class="level3">
<h3 class="anchored" data-anchor-id="plot-annotations">Plot Annotations</h3>
<section id="plot_annotations" class="level4">
<h4 class="anchored" data-anchor-id="plot_annotations"><code>plot_annotations</code></h4>
<ul>
<li><strong>Description</strong>: Metadata fields used to color-code plots.</li>
<li><strong>Example</strong>: <code>Genotype</code></li>
<li><strong>Usage</strong>: Specify metadata that can be used to annotate or differentiate groups in visualizations, such as PCA or clustering plots.</li>
</ul>
</section>
</section>
<section id="quality-control-qc" class="level3">
<h3 class="anchored" data-anchor-id="quality-control-qc">Quality Control (QC)</h3>
<section id="qc" class="level4">
<h4 class="anchored" data-anchor-id="qc"><code>qc</code></h4>
<p>This section contains quality control thresholds to filter the dataset.</p>
<ul>
<li><strong><code>min_nsamp</code></strong>: Minimum number of samples a gene must be expressed in to be retained.
<ul>
<li><strong>Example</strong>: <code>2</code></li>
<li><strong>Usage</strong>: Genes expressed in fewer than this number of samples are removed to reduce noise.</li>
</ul></li>
<li><strong><code>min_gene_counts</code></strong>: Minimum count for a gene to be retained.
<ul>
<li><strong>Example</strong>: <code>1</code></li>
<li><strong>Usage</strong>: Genes with fewer counts than this threshold will be filtered out. This helps remove low-expression, noisy genes.</li>
</ul></li>
</ul>
</section>
</section>
<section id="clustering-configuration" class="level3">
<h3 class="anchored" data-anchor-id="clustering-configuration">Clustering Configuration</h3>
<section id="exp_cluster" class="level4">
<h4 class="anchored" data-anchor-id="exp_cluster"><code>exp_cluster</code></h4>
<ul>
<li><strong><code>k</code></strong>: Number of clusters for unsupervised clustering of expression data.
<ul>
<li><strong>Example</strong>: <code>2</code></li>
<li><strong>Usage</strong>: Defines how many groups the samples will be clustered into based on expression patterns.</li>
</ul></li>
</ul>
</section>
<section id="metadata_clusters" class="level4">
<h4 class="anchored" data-anchor-id="metadata_clusters"><code>metadata_clusters</code></h4>
<ul>
<li><strong>Description</strong>: Controls clustering based on metadata scores, like pathway activity or microenvironment composition.</li>
<li><strong><code>pathway_scores</code></strong>: Defines the number of clusters (<code>k</code>) for clustering samples based on pathway activity scores.
<ul>
<li><strong>Example</strong>: <code>k: 2</code></li>
</ul></li>
<li><strong><code>microenv_scores</code></strong>: Defines the number of clusters (<code>k</code>) for clustering based on microenvironment scores.
<ul>
<li><strong>Example</strong>: <code>k: 3</code></li>
</ul></li>
</ul>
</section>
</section>
<section id="pathway-collections" class="level3">
<h3 class="anchored" data-anchor-id="pathway-collections">Pathway Collections</h3>
<section id="pathway_collections" class="level4">
<h4 class="anchored" data-anchor-id="pathway_collections"><code>pathway_collections</code></h4>
<ul>
<li><strong>Description</strong>: Defines which pathway collections to use for pathway activity scoring.</li>
<li><strong>Example</strong>: <code>CP:KEGG</code></li>
<li><strong>Usage</strong>: Here, the <code>CP:KEGG</code> collection from MSigDB is used. You can choose different collections based on your research question.</li>
</ul>
</section>
</section>
<section id="targeted-visualizations" class="level3">
<h3 class="anchored" data-anchor-id="targeted-visualizations">Targeted Visualizations</h3>
<p>This section defines genes and pathways of interest for visualization.</p>
<section id="heatmap_genes" class="level4">
<h4 class="anchored" data-anchor-id="heatmap_genes"><code>heatmap_genes</code></h4>
<ul>
<li><p><strong>Description</strong>: Genes or pathways to include in heatmaps.</p></li>
<li><p><strong>Example</strong>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode yaml code-with-copy"><code class="sourceCode yaml"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">[</span><span class="at">Exo1</span><span class="kw">,</span><span class="at"> Lig1</span><span class="kw">,</span><span class="at"> Mlh1</span><span class="kw">,</span><span class="at"> Mlh3</span><span class="kw">,</span><span class="at"> Msh2</span><span class="kw">,</span><span class="at"> Msh3</span><span class="kw">,</span><span class="at"> Msh6</span><span class="kw">,</span><span class="at"> Pcna</span><span class="kw">,</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="at">   Pms2</span><span class="kw">,</span><span class="at"> Pold1</span><span class="kw">,</span><span class="at"> Pold2</span><span class="kw">,</span><span class="at"> Pold3</span><span class="kw">,</span><span class="at"> Pold4</span><span class="kw">,</span><span class="at"> Rfc1</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">,</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="at">   Rfc4</span><span class="kw">,</span><span class="at"> Rfc5</span><span class="kw">,</span><span class="at"> Rpa1</span><span class="kw">,</span><span class="at"> Rpa2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">,</span><span class="at"> Ssbp1</span><span class="kw">],</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="at">  KEGG_CITRATE_CYCLE_TCA_CYCLE</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="kw">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
<li><p><strong>Usage</strong>: Provide either a list of specific genes or pathways to be used for heatmap visualizations. This allows you to focus on specific areas of interest.</p></li>
</ul>
</section>
<section id="heatmap_pathways" class="level4">
<h4 class="anchored" data-anchor-id="heatmap_pathways"><code>heatmap_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pathways to include in heatmap visualizations.</li>
<li><strong>Example</strong>: <code>[KEGG_CITRATE_CYCLE_TCA_CYCLE, KEGG_MISMATCH_REPAIR]</code></li>
<li><strong>Usage</strong>: Specify pathways to visualize their activity across samples.</li>
</ul>
</section>
<section id="boxplot_genes" class="level4">
<h4 class="anchored" data-anchor-id="boxplot_genes"><code>boxplot_genes</code></h4>
<ul>
<li><strong>Description</strong>: Genes to visualize using boxplots.</li>
<li><strong>Example</strong>: <code>[Msh2, Rpa3]</code></li>
<li><strong>Usage</strong>: Define genes for which you want to generate boxplots to compare their expression across conditions.</li>
</ul>
</section>
<section id="boxplot_pathways" class="level4">
<h4 class="anchored" data-anchor-id="boxplot_pathways"><code>boxplot_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pathways to visualize using boxplots.</li>
<li><strong>Example</strong>: <code>[KEGG_OXIDATIVE_PHOSPHORYLATION, KEGG_MISMATCH_REPAIR]</code></li>
<li><strong>Usage</strong>: Provide pathways to compare their activity across different experimental conditions using boxplots.</li>
</ul>
</section>
<section id="correlation_genes" class="level4">
<h4 class="anchored" data-anchor-id="correlation_genes"><code>correlation_genes</code></h4>
<ul>
<li><strong>Description</strong>: Pairs of genes to visualize for correlation analysis.</li>
<li><strong>Example</strong>: <code>[[Msh2, Rfc3], [Msh2, Rfc2]]</code></li>
<li><strong>Usage</strong>: Specify pairs of genes to plot correlations between their expression values, helping to reveal co-regulation or interaction.</li>
</ul>
</section>
<section id="correlation_pathways" class="level4">
<h4 class="anchored" data-anchor-id="correlation_pathways"><code>correlation_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pairs of pathways to visualize for correlation analysis.</li>
<li><strong>Example</strong>: <code>[]</code> (Empty if none)</li>
<li><strong>Usage</strong>: Define pairs of pathways to examine correlations between their activity. Leave empty if not applicable.</li>
</ul>
</section>
</section>
</section>
<section id="example-params.yml-file" class="level2">
<h2 class="anchored" data-anchor-id="example-params.yml-file">Example <code>params.yml</code> File</h2>
<p>Below is an example configuration for <code>params.yml</code>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode yaml code-with-copy"><code class="sourceCode yaml"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">project</span><span class="kw">:</span><span class="at"> Mouse STING KO</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data_file</span><span class="kw">:</span><span class="at"> data/STING_KO/exp_data.RDS</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">use_symbol</span><span class="kw">:</span><span class="at"> </span><span class="ch">true</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">show_code</span><span class="kw">:</span><span class="at"> </span><span class="ch">false</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">species</span><span class="kw">:</span><span class="at"> Mus musculus</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_annotations</span><span class="kw">:</span><span class="at"> Genotype</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qc</span><span class="kw">:</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_nsamp</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_gene_counts</span><span class="kw">:</span><span class="at"> </span><span class="dv">1</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">exp_cluster</span><span class="kw">:</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata_clusters</span><span class="kw">:</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">pathway_scores</span><span class="kw">:</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">microenv_scores</span><span class="kw">:</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">3</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pathway_collections</span><span class="kw">:</span><span class="at"> CP:KEGG</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">[</span><span class="at">Exo1</span><span class="kw">,</span><span class="at"> Lig1</span><span class="kw">,</span><span class="at"> Mlh1</span><span class="kw">,</span><span class="at"> Mlh3</span><span class="kw">,</span><span class="at"> Msh2</span><span class="kw">,</span><span class="at"> Msh3</span><span class="kw">,</span><span class="at"> Msh6</span><span class="kw">,</span><span class="at"> Pcna</span><span class="kw">,</span><span class="at"> Pms2</span><span class="kw">,</span><span class="at"> Pold1</span><span class="kw">,</span><span class="at"> Pold2</span><span class="kw">,</span><span class="at"> Pold3</span><span class="kw">,</span><span class="at"> Pold4</span><span class="kw">,</span><span class="at"> Rfc1</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">,</span><span class="at"> Rfc4</span><span class="kw">,</span><span class="at"> Rfc5</span><span class="kw">,</span><span class="at"> Rpa1</span><span class="kw">,</span><span class="at"> Rpa2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">,</span><span class="at"> Ssbp1</span><span class="kw">],</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="at">  KEGG_CITRATE_CYCLE_TCA_CYCLE</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">]</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">KEGG_CITRATE_CYCLE_TCA_CYCLE</span><span class="kw">,</span><span class="at"> KEGG_MISMATCH_REPAIR</span><span class="kw">]</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">]</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">KEGG_OXIDATIVE_PHOSPHORYLATION</span><span class="kw">,</span><span class="at"> KEGG_MISMATCH_REPAIR</span><span class="kw">]</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="fu">correlation_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">],</span><span class="at"> </span><span class="kw">[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">]]</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="fu">correlation_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="editing-tips" class="level2">
<h2 class="anchored" data-anchor-id="editing-tips">Editing Tips</h2>
<ol type="1">
<li><strong>YAML Formatting</strong>: YAML is indentation-sensitive, so make sure to use consistent spacing to avoid syntax errors.</li>
<li><strong>Commenting</strong>: You can add comments using <code>#</code> to document specific settings or notes.</li>
<li><strong>Validation</strong>: Double-check file paths, gene names, and pathway names to ensure they match the data to avoid errors during analysis.</li>
</ol>
</section>
<section id="common-issues" class="level2">
<h2 class="anchored" data-anchor-id="common-issues">Common Issues</h2>
<ul>
<li><strong>Incorrect Paths</strong>: Make sure the <code>data_file</code> path is correct so the program can locate the dataset.</li>
<li><strong>Missing Values</strong>: Parameters must be properly filled in; any missing required parameters can cause the analysis to fail.</li>
<li><strong>Formatting Errors</strong>: Incorrect indentation in YAML can lead to parsing issues—be mindful of spacing.</li>
</ul>
<p>Feel free to use this guide to understand and adjust the parameters for your specific RNA-seq analysis needs. If you have any questions or need further assistance, please reach out to your analysis team.</p>
<p>This <code>params.yml</code> file is used to configure the unsupervised RNA-seq analysis pipeline. It allows users to specify settings for clustering, filtering, normalization, and visualization of their RNA-seq dataset. Below, each parameter is explained in detail to help guide you in filling out or understanding this configuration.</p>
</section>
<section id="structure-of-params.yml-1" class="level2">
<h2 class="anchored" data-anchor-id="structure-of-params.yml-1">Structure of <code>params.yml</code></h2>
<p>The file is formatted in YAML, which makes it easy to read and edit. Below, we break down the purpose of each parameter.</p>
<section id="project-information-1" class="level3">
<h3 class="anchored" data-anchor-id="project-information-1">Project Information</h3>
<section id="project-1" class="level4">
<h4 class="anchored" data-anchor-id="project-1"><code>project</code></h4>
<ul>
<li><strong>Description</strong>: A descriptive name for the project.</li>
<li><strong>Example</strong>: <code>Mouse STING KO</code></li>
<li><strong>Usage</strong>: This will be used as the title of the analysis report to identify the project.</li>
</ul>
</section>
<section id="data_file-1" class="level4">
<h4 class="anchored" data-anchor-id="data_file-1"><code>data_file</code></h4>
<ul>
<li><strong>Description</strong>: Path to the RNA-seq data file in <code>.RDS</code> format.</li>
<li><strong>Example</strong>: <code>data/STING_KO/exp_data.RDS</code></li>
<li><strong>Usage</strong>: This should point to the expression data file that you wish to analyze. The <code>.RDS</code> file contains your pre-processed gene expression matrix.</li>
</ul>
</section>
<section id="use_symbol-1" class="level4">
<h4 class="anchored" data-anchor-id="use_symbol-1"><code>use_symbol</code></h4>
<ul>
<li><strong>Description</strong>: Whether to convert gene identifiers to gene symbols.</li>
<li><strong>Options</strong>: <code>true</code> or <code>false</code></li>
<li><strong>Example</strong>: <code>true</code></li>
<li><strong>Usage</strong>: Set to <code>true</code> to rebase Ensembl IDs to gene symbols, which can make downstream analysis and interpretation easier for biological studies.</li>
</ul>
</section>
<section id="show_code-1" class="level4">
<h4 class="anchored" data-anchor-id="show_code-1"><code>show_code</code></h4>
<ul>
<li><strong>Description</strong>: Whether to display the code in the output report.</li>
<li><strong>Options</strong>: <code>true</code> or <code>false</code></li>
<li><strong>Example</strong>: <code>false</code></li>
<li><strong>Usage</strong>: Set this to <code>true</code> if you want to include R code in the final output report, which may help for reproducibility or educational purposes.</li>
</ul>
</section>
</section>
<section id="species-information-1" class="level3">
<h3 class="anchored" data-anchor-id="species-information-1">Species Information</h3>
<section id="species-1" class="level4">
<h4 class="anchored" data-anchor-id="species-1"><code>species</code></h4>
<ul>
<li><strong>Description</strong>: Specifies the species of the dataset.</li>
<li><strong>Example</strong>: <code>Mus musculus</code></li>
<li><strong>Usage</strong>: This parameter helps select the appropriate gene sets and pathways for the analysis. Here, <code>Mus musculus</code> (mouse) is specified.</li>
</ul>
</section>
</section>
<section id="plot-annotations-1" class="level3">
<h3 class="anchored" data-anchor-id="plot-annotations-1">Plot Annotations</h3>
<section id="plot_annotations-1" class="level4">
<h4 class="anchored" data-anchor-id="plot_annotations-1"><code>plot_annotations</code></h4>
<ul>
<li><strong>Description</strong>: Metadata fields used to color-code plots.</li>
<li><strong>Example</strong>: <code>Genotype</code></li>
<li><strong>Usage</strong>: Specify metadata that can be used to annotate or differentiate groups in visualizations, such as PCA or clustering plots.</li>
</ul>
</section>
</section>
<section id="quality-control-qc-1" class="level3">
<h3 class="anchored" data-anchor-id="quality-control-qc-1">Quality Control (QC)</h3>
<section id="qc-1" class="level4">
<h4 class="anchored" data-anchor-id="qc-1"><code>qc</code></h4>
<p>This section contains quality control thresholds to filter the dataset.</p>
<ul>
<li><strong><code>min_nsamp</code></strong>: Minimum number of samples a gene must be expressed in to be retained.
<ul>
<li><strong>Example</strong>: <code>2</code></li>
<li><strong>Usage</strong>: Genes expressed in fewer than this number of samples are removed to reduce noise.</li>
</ul></li>
<li><strong><code>min_gene_counts</code></strong>: Minimum count for a gene to be retained.
<ul>
<li><strong>Example</strong>: <code>1</code></li>
<li><strong>Usage</strong>: Genes with fewer counts than this threshold will be filtered out. This helps remove low-expression, noisy genes.</li>
</ul></li>
</ul>
</section>
</section>
<section id="clustering-configuration-1" class="level3">
<h3 class="anchored" data-anchor-id="clustering-configuration-1">Clustering Configuration</h3>
<section id="exp_cluster-1" class="level4">
<h4 class="anchored" data-anchor-id="exp_cluster-1"><code>exp_cluster</code></h4>
<ul>
<li><strong><code>k</code></strong>: Number of clusters for unsupervised clustering of expression data.
<ul>
<li><strong>Example</strong>: <code>2</code></li>
<li><strong>Usage</strong>: Defines how many groups the samples will be clustered into based on expression patterns.</li>
</ul></li>
</ul>
</section>
<section id="metadata_clusters-1" class="level4">
<h4 class="anchored" data-anchor-id="metadata_clusters-1"><code>metadata_clusters</code></h4>
<ul>
<li><strong>Description</strong>: Controls clustering based on metadata scores, like pathway activity or microenvironment composition.</li>
<li><strong><code>pathway_scores</code></strong>: Defines the number of clusters (<code>k</code>) for clustering samples based on pathway activity scores.
<ul>
<li><strong>Example</strong>: <code>k: 2</code></li>
</ul></li>
<li><strong><code>microenv_scores</code></strong>: Defines the number of clusters (<code>k</code>) for clustering based on microenvironment scores.
<ul>
<li><strong>Example</strong>: <code>k: 3</code></li>
</ul></li>
</ul>
</section>
</section>
<section id="pathway-collections-1" class="level3">
<h3 class="anchored" data-anchor-id="pathway-collections-1">Pathway Collections</h3>
<section id="pathway_collections-1" class="level4">
<h4 class="anchored" data-anchor-id="pathway_collections-1"><code>pathway_collections</code></h4>
<ul>
<li><strong>Description</strong>: Defines which pathway collections to use for pathway activity scoring.</li>
<li><strong>Example</strong>: <code>CP:KEGG</code></li>
<li><strong>Usage</strong>: Here, the <code>CP:KEGG</code> collection from MSigDB is used. You can choose different collections based on your research question.</li>
</ul>
</section>
</section>
<section id="targeted-visualizations-1" class="level3">
<h3 class="anchored" data-anchor-id="targeted-visualizations-1">Targeted Visualizations</h3>
<p>This section defines genes and pathways of interest for visualization.</p>
<section id="heatmap_genes-1" class="level4">
<h4 class="anchored" data-anchor-id="heatmap_genes-1"><code>heatmap_genes</code></h4>
<ul>
<li><p><strong>Description</strong>: Genes or pathways to include in heatmaps.</p></li>
<li><p><strong>Example</strong>:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode yaml code-with-copy"><code class="sourceCode yaml"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">[</span><span class="at">Exo1</span><span class="kw">,</span><span class="at"> Lig1</span><span class="kw">,</span><span class="at"> Mlh1</span><span class="kw">,</span><span class="at"> Mlh3</span><span class="kw">,</span><span class="at"> Msh2</span><span class="kw">,</span><span class="at"> Msh3</span><span class="kw">,</span><span class="at"> Msh6</span><span class="kw">,</span><span class="at"> Pcna</span><span class="kw">,</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="at">   Pms2</span><span class="kw">,</span><span class="at"> Pold1</span><span class="kw">,</span><span class="at"> Pold2</span><span class="kw">,</span><span class="at"> Pold3</span><span class="kw">,</span><span class="at"> Pold4</span><span class="kw">,</span><span class="at"> Rfc1</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">,</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="at">   Rfc4</span><span class="kw">,</span><span class="at"> Rfc5</span><span class="kw">,</span><span class="at"> Rpa1</span><span class="kw">,</span><span class="at"> Rpa2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">,</span><span class="at"> Ssbp1</span><span class="kw">],</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="at">  KEGG_CITRATE_CYCLE_TCA_CYCLE</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="kw">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
<li><p><strong>Usage</strong>: Provide either a list of specific genes or pathways to be used for heatmap visualizations. This allows you to focus on specific areas of interest.</p></li>
</ul>
</section>
<section id="heatmap_pathways-1" class="level4">
<h4 class="anchored" data-anchor-id="heatmap_pathways-1"><code>heatmap_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pathways to include in heatmap visualizations.</li>
<li><strong>Example</strong>: <code>[KEGG_CITRATE_CYCLE_TCA_CYCLE, KEGG_MISMATCH_REPAIR]</code></li>
<li><strong>Usage</strong>: Specify pathways to visualize their activity across samples.</li>
</ul>
</section>
<section id="boxplot_genes-1" class="level4">
<h4 class="anchored" data-anchor-id="boxplot_genes-1"><code>boxplot_genes</code></h4>
<ul>
<li><strong>Description</strong>: Genes to visualize using boxplots.</li>
<li><strong>Example</strong>: <code>[Msh2, Rpa3]</code></li>
<li><strong>Usage</strong>: Define genes for which you want to generate boxplots to compare their expression across conditions.</li>
</ul>
</section>
<section id="boxplot_pathways-1" class="level4">
<h4 class="anchored" data-anchor-id="boxplot_pathways-1"><code>boxplot_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pathways to visualize using boxplots.</li>
<li><strong>Example</strong>: <code>[KEGG_OXIDATIVE_PHOSPHORYLATION, KEGG_MISMATCH_REPAIR]</code></li>
<li><strong>Usage</strong>: Provide pathways to compare their activity across different experimental conditions using boxplots.</li>
</ul>
</section>
<section id="correlation_genes-1" class="level4">
<h4 class="anchored" data-anchor-id="correlation_genes-1"><code>correlation_genes</code></h4>
<ul>
<li><strong>Description</strong>: Pairs of genes to visualize for correlation analysis.</li>
<li><strong>Example</strong>: <code>[[Msh2, Rfc3], [Msh2, Rfc2]]</code></li>
<li><strong>Usage</strong>: Specify pairs of genes to plot correlations between their expression values, helping to reveal co-regulation or interaction.</li>
</ul>
</section>
<section id="correlation_pathways-1" class="level4">
<h4 class="anchored" data-anchor-id="correlation_pathways-1"><code>correlation_pathways</code></h4>
<ul>
<li><strong>Description</strong>: Pairs of pathways to visualize for correlation analysis.</li>
<li><strong>Example</strong>: <code>[]</code> (Empty if none)</li>
<li><strong>Usage</strong>: Define pairs of pathways to examine correlations between their activity. Leave empty if not applicable.</li>
</ul>
</section>
</section>
</section>
<section id="example-params.yml-file-1" class="level2">
<h2 class="anchored" data-anchor-id="example-params.yml-file-1">Example <code>params.yml</code> File</h2>
<p>Below is an example configuration for <code>params.yml</code>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode yaml code-with-copy"><code class="sourceCode yaml"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">project</span><span class="kw">:</span><span class="at"> Mouse STING KO</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data_file</span><span class="kw">:</span><span class="at"> data/STING_KO/exp_data.RDS</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">use_symbol</span><span class="kw">:</span><span class="at"> </span><span class="ch">true</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">show_code</span><span class="kw">:</span><span class="at"> </span><span class="ch">false</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">species</span><span class="kw">:</span><span class="at"> Mus musculus</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_annotations</span><span class="kw">:</span><span class="at"> Genotype</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qc</span><span class="kw">:</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_nsamp</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">min_gene_counts</span><span class="kw">:</span><span class="at"> </span><span class="dv">1</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">exp_cluster</span><span class="kw">:</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata_clusters</span><span class="kw">:</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">pathway_scores</span><span class="kw">:</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">2</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">microenv_scores</span><span class="kw">:</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="at">    </span><span class="fu">k</span><span class="kw">:</span><span class="at"> </span><span class="dv">3</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pathway_collections</span><span class="kw">:</span><span class="at"> CP:KEGG</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">[</span><span class="at">Exo1</span><span class="kw">,</span><span class="at"> Lig1</span><span class="kw">,</span><span class="at"> Mlh1</span><span class="kw">,</span><span class="at"> Mlh3</span><span class="kw">,</span><span class="at"> Msh2</span><span class="kw">,</span><span class="at"> Msh3</span><span class="kw">,</span><span class="at"> Msh6</span><span class="kw">,</span><span class="at"> Pcna</span><span class="kw">,</span><span class="at"> Pms2</span><span class="kw">,</span><span class="at"> Pold1</span><span class="kw">,</span><span class="at"> Pold2</span><span class="kw">,</span><span class="at"> Pold3</span><span class="kw">,</span><span class="at"> Pold4</span><span class="kw">,</span><span class="at"> Rfc1</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">,</span><span class="at"> Rfc4</span><span class="kw">,</span><span class="at"> Rfc5</span><span class="kw">,</span><span class="at"> Rpa1</span><span class="kw">,</span><span class="at"> Rpa2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">,</span><span class="at"> Ssbp1</span><span class="kw">],</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="at">  KEGG_CITRATE_CYCLE_TCA_CYCLE</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="kw">]</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">KEGG_CITRATE_CYCLE_TCA_CYCLE</span><span class="kw">,</span><span class="at"> KEGG_MISMATCH_REPAIR</span><span class="kw">]</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rpa3</span><span class="kw">]</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[</span><span class="at">KEGG_OXIDATIVE_PHOSPHORYLATION</span><span class="kw">,</span><span class="at"> KEGG_MISMATCH_REPAIR</span><span class="kw">]</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="fu">correlation_genes</span><span class="kw">:</span><span class="at"> </span><span class="kw">[[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rfc3</span><span class="kw">],</span><span class="at"> </span><span class="kw">[</span><span class="at">Msh2</span><span class="kw">,</span><span class="at"> Rfc2</span><span class="kw">]]</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="fu">correlation_pathways</span><span class="kw">:</span><span class="at"> </span><span class="kw">[]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="editing-tips-1" class="level2">
<h2 class="anchored" data-anchor-id="editing-tips-1">Editing Tips</h2>
<ol type="1">
<li><strong>YAML Formatting</strong>: YAML is indentation-sensitive, so make sure to use consistent spacing to avoid syntax errors.</li>
<li><strong>Commenting</strong>: You can add comments using <code>#</code> to document specific settings or notes.</li>
<li><strong>Validation</strong>: Double-check file paths, gene names, and pathway names to ensure they match the data to avoid errors during analysis.</li>
</ol>
</section>
<section id="common-issues-1" class="level2">
<h2 class="anchored" data-anchor-id="common-issues-1">Common Issues</h2>
<ul>
<li><strong>Incorrect Paths</strong>: Make sure the <code>data_file</code> path is correct so the program can locate the dataset.</li>
<li><strong>Missing Values</strong>: Parameters must be properly filled in; any missing required parameters can cause the analysis to fail.</li>
<li><strong>Formatting Errors</strong>: Incorrect indentation in YAML can lead to parsing issues—be mindful of spacing.</li>
</ul>
<p>Feel free to use this guide to understand and adjust the parameters for your specific RNA-seq analysis needs. If you have any questions or need further assistance, please reach out to your analysis team.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>