---
title: "Unsupervised RNA-seq analysis"
author: "Cordeliers Artificial Intelligence and Bioinformatics"
format: 
  html:
    toc: true
    embed-resources: true
editor: source
---
<img src="../man/figures/CAIBI.png" align="right" alt="" width="120" />


For the 2 examples, we are using an open dataset which is given by the `airway` package. 
It provides a gene expression dataset derived from **human bronchial epithelial cells**, treated or not with **dexamethasone** (a corticosteroid).
 
Here is an example of how the **unsupervised** part of the `CAIBIrnaseq` package can be use. If you want to use this notebook for your projects, it is available [here](https://github.com/crcordeliers/CAIBIrnaseq/blob/dev/vignettes/sup_rnaseq.Rmd)


First, we need to import the required packages :
```{r}
#| message: false
library(airway)               # dataset's package  
library(SummarizedExperiment) # Object's type we are using
library(CAIBIrnaseq)          # package
```

Before analysing the dataset, we define the variables with the names of the genes / pathways ... that we want to analyse.

These variables will change in function of how your dataset is build and which type of data it is.

If you find the definition of the variables heavy in your code, you can create a `.yml` file, for example named `params.yml`, call it in a cell, and it will automatically import the variables. Your code will be clearer (here is an [example](https://github.com/crcordeliers/CAIBIrnaseq/blob/dev/params.yml))

```{r read_params}
#| message: false

species <- "Homo sapiens"

# Annotation variable to visualize
plot_annotations <- "dex"  # treated or untreated

# Quality parameters
qc_min_nsamp <- 2
qc_min_gene_counts <- 10  

# Clustering of expression
exp_cluster <- data.frame(k = 2)

# Clustering of metadata
metadata_clusters <- list(
  pathway_scores = data.frame(k = 2),
  microenv_scores = data.frame(k = 3)
)

# The following variables are those that will need to be modified depending on the analyses you want to do  

# Pathway collections 
pathway_collections <- c("CGP", "CP", "CP:KEGG_LEGACY", "Hallmark")

# Interesting genes
heatmap_genes <- list(
  gr_response_genes <- c("FKBP5", "TSC22D3", "PER1", "ZBTB16"),
  anti_inflam_genes <- c("DUSP1", "SOCS1", "MT2A")
)

heatmap_pathways <- c(
  "DUTERTRE_ESTRADIOL_RESPONSE_24HR_DN",
  "REN_ALVEOLAR_RHABDOMYOSARCOMA_DN",
  "NUYTTEN_EZH2_TARGETS_UP",
  "PASINI_SUZ12_TARGETS_DN"
)

# Genes for the boxplots
boxplot_genes <- c("FKBP5", "TSC22D3")

# Pathways for the boxplots
boxplot_pathways <- c(
  "KUMAMOTO_RESPONSE_TO_NUTLIN_3A_UP",
  "CASTELLANO_HRAS_TARGETS_DN"
)

# Corrélations entre gènes
correlation_genes <- list(
  c("FKBP5", "TSC22D3"),
  c("FKBP5", "ZBTB16")
)

# Pathways correlation
correlation_pathways <- list(
  c("DUTERTRE_ESTRADIOL_RESPONSE_24HR_DN", "REN_ALVEOLAR_RHABDOMYOSARCOMA_DN"),
  c("REN_ALVEOLAR_RHABDOMYOSARCOMA_DN", "NUYTTEN_EZH2_TARGETS_UP")
)

```

## Load data

This section loads the RNA-seq dataset for analysis. It ensures the correct input file is used, as specified in the parameters.
rebase_gexp

Ensure your dataset is in a `Summarized Experiment` object, because all the used functions below works with SummarizedExperiment input.

If you want to know more about this type of object, please click here: [Bioconductor](https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html)

```{r load_data}
#| message: false
data(airway, package="airway")
exp_data <- airway

# If you are using your own dataset .RDS file), use this command line : 
# exp_data <- readRDS(data_file)
```

Even if the datasets are globally build the same way,  the names of the variables are not exactly the same, so if we want to keep the same code, we need to redefine a bit the variables. 

If you want to know what are the used variables in this part, run this command line : `colnames(SummarizedExperiment::rowData(exp_data))`

You should have these variables (with these exact same names):   
  - **gene_name** : The commonly used symbol or name for the gene (e.g., A1BG).    
  - **gene_id** : A unique and stable identifier for the gene, often from databases like Ensembl.  
  - **gene_length_kb** :  The length of the gene measured in kilobases   
  - **gene_description** : A brief textual summary of the gene’s function or characteristics. 
  - **gene_biotype** : A classification of the gene based on its biological function or transcript type, such as protein_coding, lncRNA, or pseudogene.  

If you are using the notebook that Clarice GROENEVELD created to convert a .xsl file into a .RDS one, you can SKIP the next cell. 
If not, you should look at how your dataset is defined. You might need to run some command line as the following ones: 

``` {r}
#| message: false
library(biomaRt)

rowData(exp_data)$gene_length_kb <- 
  (rowData(exp_data)$gene_seq_end - rowData(exp_data)$gene_seq_start) / 1000

mart <- useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl")
gene_ids <- rowData(exp_data)$gene_id

annot <- getBM(attributes = c("ensembl_gene_id", "description"),
               filters = "ensembl_gene_id",
               values = gene_ids,
               mart = mart)

matched <- match(rowData(exp_data)$gene_id, annot$ensembl_gene_id)
rowData(exp_data)$gene_description <- annot$description[matched]
```

## Pre-processing

Most datasets use ensembl gene ID by default after alignment, so this step rebases the expression data to gene names. This ensures consistency in naming for downstream analyses.

```{r rebase_gexp}
#| message: false
exp_data <- rebase_gexp(exp_data, annotation = "gene_name")
```

### Filter

Here, we filter out genes expressed in too few samples or with very low counts. This removes noise from the data and focuses on meaningful gene expressions.

```{r filter_gexp}
#| message: false
exp_data <- filter_gexp(exp_data,
                        min_nsamp = 1, 
                        min_counts = 1)
```

Visualization of the filtering process to ensure the criteria applied align with the dataset’s characteristics:

```{r plot_qc_filters, warning=FALSE}
#| message: false
colData(exp_data)$sample_id <- colnames(exp_data)
plot_qc_filters(exp_data)
```

### Normalize

Here, we apply a normalization to the expression data, making samples comparable by reducing variability due to technical differences. For datasets with few samples, `rlog` is the preferred normalization and when more samples are present, `vst` is applied.

```{r}
#| message: false
exp_data <- normalize_gexp(exp_data)
```

## PCA

Principal component analysis (PCA) identifies the major patterns in the dataset. These patterns help explore similarities or differences among samples based on gene expression.

```{r}
#| message: false
pca_res = pca_gexp(exp_data)
exp_data@metadata[["pca_res"]] <- pca_res

annotations <- setdiff(plot_annotations, c("exp_cluster", "path_cluster"))
plot_pca(exp_data, color = plot_annotations)
```

If you want something more visual, you can add a circular/oval shape to circle the different genotypes of samples, use the `fviz_pca_ind` function from the `factoextra` package.
With this dataset, it is not relevant but it can be with your persona dataset.
Here, it highlights the fact that the 2 groups are not crossing each other. The `trt` group has more PC1, whereas the `untrt` group has less.
We could conclude that PC1 is more represented in the treated samples.

```{r, message=FALSE}
library(factoextra)
groups <- SummarizedExperiment::colData(exp_data)$dex  # here we want to split in function of `treated` and `untreated`

fviz_pca_ind(pca_res,
             geom = "point",
             habillage = groups,
             palette = c("#00AFBB", "#E7B800"),  # Personalized colors
             addEllipses = TRUE,
             ellipse.type = "confidence",
             repel = TRUE,
             label = "none"
)
```


## Unsupervised clustering


Here, we group samples based on expression patterns without prior knowledge using hierarchical clustering on either a selected gene list from the parameters or, by default, the 2000 most highly expressed genes.

This can be useful for discovering sample subgroups or new biological insights.

```{r}
#| message: false
#| label: exp cluster
#| eval: !expr is.list(exp_cluster)
exp_data <- cluster_exp(exp_data, k = exp_cluster$k, genes = exp_cluster$genes, n_pcs = 3)
```

Visual representation of expression levels for HVG across clusters, highlighting distinct patterns.

```{r}
#| label: plot hvg
#| fig.width: 8
#| fig.height: 5
#| message: false
hvg <- highly_variable_genes(exp_data)
exp_cluster <- data.frame(k = 2)
hm <- plot_exp_heatmap(exp_data,  genes = hvg, 
                 annotations = c(plot_annotations, "exp_cluster"),
                 show_rownames = FALSE,
                 hm_color_limits = c(-2,2),
                 fname = "results/unsup/clustering/heatmap_2000hvg_exp_cluster.pdf")
hm
```


## Pathway activity

Pathway analysis enables us to understand the functional implications of gene expression changes. Here, we analyze the dataset for pathway activity using two methods.

### PROGENy

PROGENy is a collection of only 14 core pathway responsive genes from large signaling perturbation experiments. For more information see the [original paper](https://www.nature.com/articles/s41467-017-02391-6).

The returned plot will give us information about the pathways that are activated for each sample. There is especially one pathway that is highly activated : `EGFR` , in the sample `SRR1039517`

```{r}
progeny_scores <- score_progeny(exp_data, species = "Homo sapiens")


```

```{r, warning=FALSE}
#| message: false
#| label: progenyprovided
progeny_scores <- score_progeny(exp_data, species = "Homo sapiens")

metadata(exp_data)[["progeny_scores"]] <- progeny_scores   

plot_progeny_heatmap(exp_data, annotations = plot_annotations,
                     fname = "results/unsup/pathways/hm_progeny_scores.pdf")
write.csv(progeny_scores, file = "results/unsup/pathways/progeny_scores.csv")
```


### Pathways
Ensure your dataset is in a `Summarized Experiment` object, because all the used functions below works with SummarizedExperiment input.

Pathway collections available in the MSIGdb can be specified in the parameters. These pathways are scored and ranked by their variance in the data. These are the available collections (use `gs_subcollection` as name except for Hallmarks, which should be 'H').

```{r, warning=FALSE}
#| message: false
#| label: collections
library(msigdbr)
library(dplyr)
library(kableExtra)


msigdbr::msigdbr_collections() |>
  kableExtra::kbl() |>
  kableExtra::kable_styling() |>
  kableExtra::scroll_box(height = "300px")

```


```{r, warning=FALSE}
#| message: false
#| label: pathways
#| fig-width: 10
#| fig-height: 5

pathways <- get_annotation_collection(pathway_collections, 
                                      species = species)
pathway_scores <- score_pathways(exp_data, pathways, verbose = FALSE)
metadata(exp_data)[["pathway_scores"]] <- pathway_scores

collections <- pathway_collections |> 
  paste(collapse = "_") |>
  stringr::str_remove("\\:")

plot_pathway_heatmap(exp_data, annotations = plot_annotations,
                    fwidth = 9,
                    fname = stringr::str_glue(
                      "results/unsup/pathways/hm_paths_{collections}_top20.pdf")
                    )
write.csv(pathways, file = stringr::str_glue("results/unsup/pathways/paths_{collections}.csv"))
```


## Microenvironment scores

This step calculates immune and stromal cell type abundances using MCPcounter or mMCPcounter. It helps to infer the composition of the tumor microenvironment or similar contexts.

```{r, warning=FALSE}
#| message: false
#| label: microenvironment
mcp_scores <- mcp_counter(exp_data, species = species)
S4Vectors::metadata(exp_data)[["microenv_scores"]] <- mcp_scores

plot_microenv_heatmap(exp_data, annotations = c("dex", "exp_cluster"),
                      fname = "results/unsup/tme/heatSmap_mcpcounter.pdf")
write.csv(mcp_scores, file = "results/unsup/tme/scores_mcpcounter.csv")
```
By default, the rows are order by scores. But, the `plot_microenv_heatsmap` function has the `ellipsis argument`. That means that this function can have a wide range of inputs. 
So, it is possible to plot the rows in a different order than the default one :

```{r, warning=FALSE, message=FALSE}
plot_microenv_heatmap(exp_data,
                      annotations = c("dex", "exp_cluster"),
                      fname = "results/unsup/tme/heatmap_sorted_bydex.pdf",
                      cluster_rows = FALSE)
```

## Targeted plots

This section focuses on visualizing specific genes or pathways of interest, as specified in the parameters.

### Heatmaps

Generates heatmaps for pre-selected genes of interest to observe their expression across samples or conditions.


```{r}
#| message: false
#| label: plot_hm_genes
#| fig-width: 10
#| fig-height: 4

hms <- lapply(1:length(heatmap_genes), function(i) {
  gene_annot <- SummarizedExperiment::rowData(exp_data)
  genes <- heatmap_genes[[i]]
  name <- ifelse(is.null(names(heatmap_genes)), i, names(heatmap_genes)[i])
  plot_exp_heatmap(exp_data, genes = genes, 
                   annotations = plot_annotations,
                   fname = stringr::str_glue("results/unsup/targeted/hm_genes_{i}.pdf"))
})
patchwork::wrap_plots(hms, ncol = 2, guides = "collect")
```

#### Selected pathways

```{r, warning=FALSE}
#| message: false
#| label: plot_hm_pathways
#| fig-width: 7
#| fig-height: 4

valid_pathways <- intersect(heatmap_pathways, rownames(pathway_scores))

plot_pathway_heatmap(exp_data, 
                     annotations = plot_annotations, 
                     pathways = valid_pathways,
                     fname = stringr::str_glue("results/unsup/targeted/hm_pathways_selected.pdf"))
```

### Boxplots

Boxplots provide a clear comparison of expression levels across experimental groups or conditions.
All jobs still running at 10:00 on Friday will be killed as part of the maintenance.
#### Selected genes

```{r}
#| message: false
#| label: plot_box_genes
#| warning: false
genes <- boxplot_genes
annotations <- plot_annotations

boxplots <- lapply(genes, function(gene) {
  lapply(annotations, function(annotation) {
    plt <- plot_exp_boxplot(exp_data, gene = gene, 
                   annotation = annotation, 
                   color_var = annotation, 
                   pt_size = 2,
                   fname = stringr::str_glue("results/unsup/targeted/boxplots/box_{gene}_{annotation}.pdf"))
  })
}) |> purrr::flatten()
```

```{r}
#| message: false
#| label: plot_box_genes2
#| fig-width: 10
#| fig-height: 4
patchwork::wrap_plots(boxplots, nrows = round(length(boxplots)/2), guides = "collect")
```

#### Selected pathways

```{r}
#| label: plot_box_paths
#| message: false
#| warning: false
paths <- boxplot_pathways
annotations <- plot_annotations

boxplots <- lapply(paths, function(path) {
  lapply(annotations, function(annotation) {
    plt <- plot_path_boxplot(exp_data, 
                             pathway = path,
                   annotation = annotation, 
                   color_var = annotation, 
                   pt_size = 2,
                   fname = stringr::str_glue("results/unsup/targeted/boxplots/box_{path}_{annotation}.pdf"))
  })
}) |> purrr::flatten()
```

```{r}
#| label: plot_box_paths2
#| fig-width: 10
#| fig-height: 4
#| message: false
patchwork::wrap_plots(boxplots, nrows = round(length(boxplots)/2), guides = "collect")
```

### Correlations

This section visualizes relationships between pairs of genes or pathways by plotting their expression/activity correlations. Correlation analysis can reveal important co-regulation or interaction patterns, helping to uncover biologically meaningful relationships.

#### Selected genes

Here we plot the correlation between selected gene pairs across the dataset. Each pair is plotted separately, and color-coded by sample annotation.

```{r}
#| message: false
#| label: plot_correlation_genes

gene_pairs <- correlation_genes
annotations <- plot_annotations

cor_plts <- lapply(gene_pairs, function(gene_pair) {
  lapply(annotations, function(annot) {
    # Correction de la construction du nom de fichier
    fname <- stringr::str_glue("results/unsup/targeted/correlations/cor_{gene_pair[1]}_{gene_pair[2]}_color={annot}.pdf")
    
    # Génération du scatter plot
    plot_exp_scatter(exp_data, 
                     gene1 = gene_pair[1],
                     gene2 = gene_pair[2], 
                     color_var = annot,
                     fname = fname)
  })
}) |> purrr::flatten()
```

```{r}
#| message: false
#| label: plot_correlation_genes2
#| fig-width: 10
#| fig-height: 4
patchwork::wrap_plots(cor_plts, nrows = round(length(cor_plts)/2), guides = "collect")
```

#### Selected pathways

Correlation plots for selected pathways can help identify similarities or differences in pathway activity patterns across samples. Each pathway pair is plotted separately and color-coded by sample annotation to illustrate trends within each condition.

```{r}
#| message: false
#| label: plot_correlation_paths

path_pairs <-correlation_pathways
annotations <- plot_annotations

cor_plts <- lapply(path_pairs, function(path_pair) {
  lapply(annotations, function(annot) {
      plot_path_scatter(exp_data, 
                   pathway1 = path_pair[1],
                   pathway2 = path_pair[2],## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was 
                   color_var = annot,
                   fname = stringr::str_glue(
                     "results/unsup/targeted/correlations/cor_{path_pair[1]}_{path_pair[2]}_color={annot}.pdf"))
  })
}) |> purrr::flatten()
```

```{r}
#| message: false
#| label: plot_correlation_paths2
#| fig-width: 10
#| fig-height: 4
patchwork::wrap_plots(cor_plts, nrows = round(length(cor_plts)/2), guides = "collect")
```

## Cluster using metadata

```{r}
#| message: false
#| label: cluster_metadata
#| warning: false
types = names(metadata_clusters)

for(type in types) {
  exp_data <- cluster_metadata(exp_data, 
                   metadata_name = type, 
                   k = metadata_clusters[[type]]$k, 
                   features = metadata_clusters[[type]]$features,
                   n_pcs = 3 )
}
```

## Save SummarizedExperiment

The final step saves the processed dataset and results. This ensures all outputs can be revisited or shared for further analysis.

```{r}
#| message: false
saveRDS(exp_data, file = stringr::str_glue("results/unsup/data_SummarizedExp_{lubridate::today()}.RDS"))
```

## Report parameters

For reproducibility, the parameters used in the analysis and the computational environment details are documented.

### sessionInfo

The `sessionInfo()` prints out all packages loaded at the time of analysis, as well as their versions.

```{r}
sessionInfo()
```

