<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pathview for KEGG • CAIBIrnaseq</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pathview for KEGG">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CAIBIrnaseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation Guide</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">References</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-examples" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-examples">
<li><a class="dropdown-item" href="../articles/unsup_rnaseq.html">Unsupervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/sup_rnaseq.html">Supervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/venn.html">Comparing datasets</a></li>
    <li><a class="dropdown-item" href="../articles/pathview.html">Pathview</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pathview for KEGG</h1>
                        <h4 data-toc-skip class="author">Cordeliers
Artificial Intelligence and Bioinformatics</h4>
            
            <h4 data-toc-skip class="date">2025-05-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/crcordeliers/CAIBIrnaseq.git/blob/HEAD/vignettes/pathview.Rmd" class="external-link"><code>vignettes/pathview.Rmd</code></a></small>
      <div class="d-none name"><code>pathview.Rmd</code></div>
    </div>

    
    
<p><img src="../reference/figures/CAIBI.png" align="right" alt="" width="120"></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#|message: false</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: generics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'generics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span></span>
<span><span class="co">##     setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,</span></span>
<span><span class="co">##     unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/datapplab/pathview" class="external-link">pathview</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## ##############################################################################</span></span>
<span><span class="co">## Pathview is an open source software package distributed under GNU General</span></span>
<span><span class="co">## Public License version 3 (GPLv3). Details of GPLv3 is available at</span></span>
<span><span class="co">## http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to</span></span>
<span><span class="co">## formally cite the original Pathview paper (not just mention it) in publications</span></span>
<span><span class="co">## or products. For details, do citation("pathview") within R.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG</span></span>
<span><span class="co">## license agreement (details at http://www.kegg.jp/kegg/legal.html).</span></span>
<span><span class="co">## ##############################################################################</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/" class="external-link">knitr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gse"</span><span class="op">)</span></span>
<span><span class="va">macrophage</span> <span class="op">&lt;-</span> <span class="va">gse</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></span>
<span><span class="va">macrophage</span> <span class="op">&lt;-</span> <span class="va">macrophage</span><span class="op">[</span>,<span class="va">macrophage</span><span class="op">$</span><span class="va">condition</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"naive"</span>, <span class="st">"IFNg"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"\\..*$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"\\..*$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S'assurer que la variable condition_name est bien un facteur</span></span>
<span><span class="va">macrophage</span><span class="op">$</span><span class="va">condition_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">macrophage</span><span class="op">$</span><span class="va">condition_name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Créer l'objet DESeq2 et exécuter l'analyse</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">macrophage</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">condition_name</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## using counts and average transcript lengths from tximeta</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## estimating size factors</span></span></code></pre>
<pre><code><span><span class="co">## using 'avgTxLength' from assays(dds), correcting for library size</span></span></code></pre>
<pre><code><span><span class="co">## estimating dispersions</span></span></code></pre>
<pre><code><span><span class="co">## gene-wise dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## mean-dispersion relationship</span></span></code></pre>
<pre><code><span><span class="co">## final dispersion estimates</span></span></code></pre>
<pre><code><span><span class="co">## fitting model and testing</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Obtenir les résultats différentiels</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Vérifier les identifiants de gènes (par défaut ce sont des Ensembl IDs)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span>  <span class="co"># par exemple ENSG00000000003</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419" "ENSG00000000457"</span></span>
<span><span class="co">## [5] "ENSG00000000460" "ENSG00000000938"</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Mapper les Ensembl IDs vers Entrez IDs</span></span>
<span><span class="va">ens_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="va">entrez_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">ens_ids</span>,</span>
<span>                     column <span class="op">=</span> <span class="st">"ENTREZID"</span>, keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>, multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ajouter les Entrez IDs à l'objet res</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">entrez</span> <span class="op">&lt;-</span> <span class="va">entrez_ids</span></span>
<span></span>
<span><span class="co"># Supprimer les NA (pour garder uniquement les gènes ayant un Entrez ID)</span></span>
<span><span class="va">res_clean</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">entrez</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># Créer un vecteur nommé de log2FoldChange avec les identifiants Entrez</span></span>
<span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="va">res_clean</span><span class="op">$</span><span class="va">log2FoldChange</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">res_clean</span><span class="op">$</span><span class="va">entrez</span></span>
<span></span>
<span><span class="co"># Visualiser sur une voie KEGG, par exemple hsa04110 (voie d'apoptose)</span></span>
<span><span class="co"># Vous pouvez changer le pathway.id selon le pathway que vous voulez visualiser</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pathview/man/pathview.html" class="external-link">pathview</a></span><span class="op">(</span></span>
<span>  gene.data <span class="op">=</span> <span class="va">gene_data</span>,          <span class="co"># Données de log2FoldChange pour les gènes DE</span></span>
<span>  pathway.id <span class="op">=</span> <span class="st">"hsa04110"</span>,        <span class="co"># ID de la voie KEGG (par ex. hsa04110 pour l'apoptose)</span></span>
<span>  species <span class="op">=</span> <span class="st">"hsa"</span>,                <span class="co"># Espèce (ici Homo sapiens)</span></span>
<span>  gene.idtype <span class="op">=</span> <span class="st">"entrez"</span>,         <span class="co"># Utiliser les identifiants Entrez</span></span>
<span>  out.suffix <span class="op">=</span> <span class="st">"macrophage_apoptosis"</span>, <span class="co"># Suffixe pour les fichiers de sortie</span></span>
<span>    low <span class="op">=</span> <span class="st">"lightgreen"</span>,                    <span class="co"># Couleur pour les valeurs faibles (logFC négatif)</span></span>
<span>  high <span class="op">=</span> <span class="st">"pink"</span>,                     <span class="co"># Couleur pour les valeurs élevées (logFC positif)</span></span>
<span>  na.color <span class="op">=</span> <span class="st">"gray"</span>,</span>
<span>  kegg.native <span class="op">=</span> <span class="cn">TRUE</span>  </span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
<pre><code><span><span class="co">## Info: Working in directory /home/marie-christian/Desktop/CAIBIrnaseq/vignettes</span></span></code></pre>
<pre><code><span><span class="co">## Info: Writing image file hsa04110.macrophage_apoptosis.png</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Afficher le graphique dans le document</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/include_graphics.html" class="external-link">include_graphics</a></span><span class="op">(</span><span class="st">"hsa04110.macrophage_apoptosis.png"</span><span class="op">)</span></span></code></pre></div>
<p><img src="hsa04110.macrophage_apoptosis.png" width="552"></p>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Clarice Groeneveld, Chloe Witz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
