<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pathview for KEGG • CAIBIrnaseq</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pathview for KEGG">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CAIBIrnaseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation Guide</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">References</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-examples" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-examples">
<li><a class="dropdown-item" href="../articles/unsup_rnaseq.html">Unsupervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/sup_rnaseq.html">Supervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/venn.html">Comparing datasets</a></li>
    <li><a class="dropdown-item" href="../articles/pathview.html">Pathview</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pathview for KEGG</h1>
                        <h4 data-toc-skip class="author">Cordeliers
Artificial Intelligence and Bioinformatics</h4>
            
            <h4 data-toc-skip class="date">2025-05-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/crcordeliers/CAIBIrnaseq.git/blob/HEAD/vignettes/pathview.Rmd" class="external-link"><code>vignettes/pathview.Rmd</code></a></small>
      <div class="d-none name"><code>pathview.Rmd</code></div>
    </div>

    
    
<p><img src="../reference/figures/CAIBI.png" align="right" alt="" width="120"></p>
<p>An additional analysis is given by the <a href="https://bioconductor.org/packages/release/bioc/html/pathview.html" class="external-link">Pathview
package</a></p>
<p><strong>Pathview</strong> is an R tool used to visualize gene
expression data on biological pathways (from KEGG), it helps for seeing
how the genes of interest are involved in known cellular processes like
the cell cycle or apoptosis.</p>
<p>To create the cell cycle graph, we use the function
<code>pathview</code>. The expected input for this function is a vector
of the <strong>log2FoldChange</strong> of the DE genes.</p>
<p>If you want to use this notebook for your projects, it is available
<a href="https://github.com/crcordeliers/CAIBIrnaseq/blob/dev/vignettes/pathview.Rmd" class="external-link">here</a></p>
<div class="section level2">
<h2 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h2>
<p>In this page, you will only see the specific code to get the graph.
The first steps are the same for every visualization. Before the next
cells of code, you will need to do all the pre-processing analyses,
until the <strong>diffexp</strong>. If you want to see the code, you can
copy/paste it from the <a href="https://crcordeliers.github.io/CAIBIrnaseq/articles/venn.html">other
pages</a></p>
<p>This is what you get after the <code>diffexp</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diffexp_mac</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##              baseMean log2FoldChange     lfcSE       pvalue         padj</span></span>
<span><span class="co">## A1BG       185.361814     0.75051222 0.3099124 4.533451e-03 2.242446e-02</span></span>
<span><span class="co">## A1BG-AS1   315.155738    -0.32048380 0.1677420 4.293224e-02 1.375238e-01</span></span>
<span><span class="co">## A1CF         1.604499     0.04585155 0.4781308 8.266114e-01           NA</span></span>
<span><span class="co">## A2M      71665.137761    -1.19356380 0.2310016 3.375187e-08 7.203143e-07</span></span>
<span><span class="co">## A2M-AS1     39.045426     0.88104208 0.5735779 1.961794e-02 7.405728e-02</span></span>
<span><span class="co">## A2ML1        2.457721    -0.04618380 0.4598400 8.400364e-01 9.379652e-01</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="conversion">Conversion<a class="anchor" aria-label="anchor" href="#conversion"></a>
</h2>
<p>The next cell will take the log2FoldChange and ENTREZID to get the
expected object for the <code>pathview</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#|message: false</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Récupérer les noms des gènes</span></span>
<span><span class="va">gene_symbols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">diffexp_mac</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Conversion symboles → Entrez IDs</span></span>
<span><span class="va">conversion</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>  <span class="va">org.Hs.eg.db</span>,</span>
<span>  keys <span class="op">=</span> <span class="va">gene_symbols</span>,</span>
<span>  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENTREZID"</span><span class="op">)</span>,</span>
<span>  keytype <span class="op">=</span> <span class="st">"SYMBOL"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Fusion avec vos données d'expression</span></span>
<span><span class="va">diffexp_mac</span><span class="op">$</span><span class="va">SYMBOL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">diffexp_mac</span><span class="op">)</span></span>
<span><span class="va">diffexp_mac</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">diffexp_mac</span>, <span class="va">conversion</span>, by <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Créer un vecteur de log2FC nommé avec les Entrez IDs</span></span>
<span><span class="va">gene.data</span> <span class="op">&lt;-</span> <span class="va">diffexp_mac</span><span class="op">$</span><span class="va">log2FoldChange</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene.data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">diffexp_mac</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span></span>
<span><span class="co"># Supprimer les entrées sans identifiants</span></span>
<span><span class="va">gene.data</span> <span class="op">&lt;-</span> <span class="va">gene.data</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene.data</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene.data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           1      503538       29974           2      144571      144568 </span></span>
<span><span class="co">##  0.75051222 -0.32048380  0.04585155 -1.19356380  0.88104208 -0.04618380</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="graph">Graph<a class="anchor" aria-label="anchor" href="#graph"></a>
</h2>
<p>Once done, we can call the function with specific parameters that
match our dataset.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#|message: false</span></span>
<span><span class="fu">pathview</span><span class="op">(</span></span>
<span>  gene.data <span class="op">=</span> <span class="va">gene.data</span>,          <span class="co"># Données de log2FoldChange pour les gènes DE</span></span>
<span>  pathway.id <span class="op">=</span> <span class="st">"hsa04110"</span>,        <span class="co"># ID de la voie KEGG (par ex. hsa04110 pour l'apoptose)</span></span>
<span>  species <span class="op">=</span> <span class="st">"hsa"</span>,                <span class="co"># Espèce (ici Homo sapiens)</span></span>
<span>  gene.idtype <span class="op">=</span> <span class="st">"entrez"</span>,         <span class="co"># Utiliser les identifiants Entrez</span></span>
<span>  out.suffix <span class="op">=</span> <span class="st">"macrophage_apoptosis"</span>, <span class="co"># Suffixe pour les fichiers de sortie</span></span>
<span>    low <span class="op">=</span> <span class="st">"lightgreen"</span>,                    <span class="co"># Couleur pour les valeurs faibles (logFC négatif)</span></span>
<span>  high <span class="op">=</span> <span class="st">"pink"</span>,                     <span class="co"># Couleur pour les valeurs élevées (logFC positif)</span></span>
<span>  na.color <span class="op">=</span> <span class="st">"gray"</span>,</span>
<span>  kegg.native <span class="op">=</span> <span class="cn">TRUE</span>  </span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="hsa04110.macrophage_apoptosis.png" width="120%" style="display: block; margin: auto;">
Using the <code>pathview package</code>, we visualized the impact of
<code>differentially expressed genes</code> from the macrophage dataset
on the KEGG apoptosis pathway (<strong>hsa04110</strong>). The pathway
diagram highlights genes based on their <strong>log2FoldChange</strong>
values: <strong>upregulated</strong> genes are shown in pink (Cdc25a,
CycA, CycB, …), while <strong>downregulated</strong> genes appear in
light green (GADD45, Cip1, CycD, …). Genes with no expression data or
missing values are colored gray. This visualization allows us to assess
how the <strong>IFN-γ treatment</strong> influences components of the
apoptotic signaling cascade, helping to identify specific regulatory
points potentially activated or suppressed in <em>response</em> to the
stimulus.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Clarice Groeneveld, Chloe Witz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
