<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Unsupervised RNA-seq analysis • CAIBIrnaseq</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Unsupervised RNA-seq analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CAIBIrnaseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation Guide</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">References</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-examples" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-examples">
<li><a class="dropdown-item" href="../articles/unsup_rnaseq.html">Unsupervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/sup_rnaseq.html">Supervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/venn.html">Comparing datasets</a></li>
    <li><a class="dropdown-item" href="../articles/pathview.html">Pathview</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="sup_rnaseq_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="sup_rnaseq_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="sup_rnaseq_files/plotly-binding-4.10.4/plotly.js"></script><script src="sup_rnaseq_files/typedarray-0.1/typedarray.min.js"></script><link href="sup_rnaseq_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="sup_rnaseq_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="sup_rnaseq_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="sup_rnaseq_files/plotly-main-2.11.1/plotly-latest.min.js"></script><script src="sup_rnaseq_files/kePrint-0.0.1/kePrint.js"></script><link href="sup_rnaseq_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Unsupervised RNA-seq analysis</h1>
                        <h4 data-toc-skip class="author">Cordeliers
Artificial Intelligence and Bioinformatics</h4>
            
            <h4 data-toc-skip class="date">2025-05-12</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/crcordeliers/CAIBIrnaseq.git/blob/HEAD/vignettes/sup_rnaseq.Rmd" class="external-link"><code>vignettes/sup_rnaseq.Rmd</code></a></small>
      <div class="d-none name"><code>sup_rnaseq.Rmd</code></div>
    </div>

    
    
<p><img src="../reference/figures/CAIBI.png" align="right" alt="" width="120"></p>
<p>For the 2 examples, we are using an open dataset which is given by
the <code>airway</code> package. It provides a gene expression dataset
derived from <strong>human bronchial epithelial cells</strong>, treated
or not with <strong>dexamethasone</strong> (a corticosteroid).</p>
<p>Here is an example of how the <strong>supervised</strong> part of the
<code>CAIBIrnaseq</code> package can be use.</p>
<p>First, we need to import the required packages :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
<p>Before analysing the dataset, we define the variables with the names
of the genes / pathways … that we want to analyse.</p>
<p>These variables will change in function of how your dataset is build
and which type of data it is.</p>
<p>If you find the definition of the variables heavy in your code, you
can create a <code>.yml</code> file, for example named
<code>params.yml</code>, call it in a cell, and it will automatically
import the variables. Your code will be clearer.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotation</span> <span class="op">&lt;-</span> <span class="st">"dex"</span></span>
<span><span class="va">diffexpMethod</span> <span class="op">&lt;-</span> <span class="st">"limma"</span> <span class="co"># Either limma or deseq2</span></span>
<span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"human"</span></span>
<span><span class="va">pathwayMethod</span> <span class="op">&lt;-</span> <span class="st">"ora"</span>  <span class="co"># We can run ORA or fgsea, we should change it at this step </span></span>
<span></span>
<span><span class="va">collections</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CGP"</span>, <span class="st">"CP"</span>, <span class="st">"CP:KEGG_LEGACY"</span>, <span class="st">"CP:BIOCARTA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pathways plus pertinents :</span></span>
<span><span class="va">pathwayOfInterest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"HALLMARK_INFLAMMATORY_RESPONSE"</span>, <span class="st">"GOBP_RESPONSE_TO_GLUCOCORTICOID"</span>,<span class="st">"GOBP_REGULATION_OF_CELL_PROLIFERATION"</span>,<span class="st">"HALLMARK_APOPTOSIS"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">boxplot_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"HALLMARK_INFLAMMATORY_RESPONSE"</span>,<span class="st">"GOBP_RESPONSE_TO_STEROID_HORMONE"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">heatmap_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  Inflammation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IL6"</span>, <span class="st">"TNF"</span>, <span class="st">"RELA"</span>, <span class="st">"NFKBIA"</span>, <span class="st">"JUN"</span>, <span class="st">"STAT3"</span><span class="op">)</span>,</span>
<span>  SteroidResponse <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NR3C1"</span>, <span class="st">"FKBP5"</span>, <span class="st">"TSC22D3"</span>, <span class="st">"ZBTB16"</span><span class="op">)</span>,</span>
<span>  Metabolism <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LDHA"</span>, <span class="st">"PDK1"</span>, <span class="st">"G6PD"</span>, <span class="st">"ACACA"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">heatmap_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"KUMAMOTO_RESPONSE_TO_NUTLIN_3A_UP"</span>,</span>
<span>  <span class="st">"BIOCARTA_SALMONELLA_PATHWAY"</span>,</span>
<span>  <span class="st">"CASTELLANO_HRAS_TARGETS_DN"</span>,</span>
<span>  <span class="st">"FARDIN_HYPOXIA_9"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Do not forget to import the package ! ;-)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://crcordeliers.github.io/CAIBIrnaseq">CAIBIrnaseq</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>This section loads the RNA-seq dataset for analysis. It ensures the
correct input file is used, as specified in the parameters.
rebase_gexp</p>
<p>Ensure your dataset is in a <code>Summarized Experiment</code>
object, because all the used functions below works with
SummarizedExperiment input.</p>
<p>If you want to know more about this type of object, please click
here: <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">Bioconductor</a></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span>, package<span class="op">=</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="va">airway</span></span></code></pre></div>
<p>Even if the datasets are globally build the same way, the names of
the variables are not exactly the same, so if we want to keep the same
code, we need to redefine a bit the variables.</p>
<p>If you want to know what are the used variables in this part, run
this command line :
<code>colnames(SummarizedExperiment::rowData(exp_data))</code></p>
<p>You should have these variables (with these exact same names):<br>
- <strong>gene_name</strong> : The commonly used symbol or name for the
gene (e.g., A1BG).<br>
- <strong>gene_id</strong> : A unique and stable identifier for the
gene, often from databases like Ensembl (e.g., ENSG00000141510).<br>
- <strong>gene_length_kb</strong> : The length of the gene measured in
kilobases<br>
- <strong>gene_description</strong> : A brief textual summary of the
gene’s function or characteristics, often pulled from annotation
databases.<br>
- <strong>gene_biotype</strong> : A classification of the gene based on
its biological function or transcript type, such as protein_coding,
lncRNA, or pseudogene.</p>
<p>If not, you should look at how your dataset is defined. You might
need to run some command line as the following ones:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_length_kb</span> <span class="op">&lt;-</span> </span>
<span>  <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_seq_end</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_seq_start</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1000</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Huber-group-EMBL/biomaRt" class="external-link">biomaRt</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Initialiser biomaRt pour Ensembl humain</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html" class="external-link">useEnsembl</a></span><span class="op">(</span><span class="st">"ensembl"</span>, dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Obtenir les descriptions basées sur les gene_id</span></span>
<span><span class="va">gene_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_gene_id"</span>, <span class="st">"description"</span><span class="op">)</span>,</span>
<span>               filters <span class="op">=</span> <span class="st">"ensembl_gene_id"</span>,</span>
<span>               values <span class="op">=</span> <span class="va">gene_ids</span>,</span>
<span>               mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Faire correspondre les descriptions aux lignes de rowData</span></span>
<span><span class="va">matched</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">annot</span><span class="op">$</span><span class="va">ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_description</span> <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">$</span><span class="va">description</span><span class="op">[</span><span class="va">matched</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h2>
<p>Most datasets use ensembl gene ID by default after alignment, so this
step rebases the expression data to gene names. This ensures consistency
in naming for downstream analyses.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rebase_gexp.html">rebase_gexp</a></span><span class="op">(</span><span class="va">exp_data</span>, annotation <span class="op">=</span> <span class="st">"gene_name"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="filter">Filter<a class="anchor" aria-label="anchor" href="#filter"></a>
</h3>
<p>Here, we filter out genes expressed in too few samples or with very
low counts. This removes noise from the data and focuses on meaningful
gene expressions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_gexp.html">filter_gexp</a></span><span class="op">(</span><span class="va">exp_data</span>,</span>
<span>                        min_nsamp <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                        min_counts <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Visualization of the filtering process to ensure the criteria applied
align with the dataset’s characteristics:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">sample_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_qc_filters.html">plot_qc_filters</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 8 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text_repel()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom2trace.default(dots[[1L]][[1L]], dots[[2L]][[1L]], dots[[3L]][[1L]]): geom_GeomTextRepel() has yet to be implemented in plotly.</span></span>
<span><span class="co">##   If you'd like to see this geom implemented,</span></span>
<span><span class="co">##   Please open an issue with your example code at</span></span>
<span><span class="co">##   https://github.com/ropensci/plotly/issues</span></span></code></pre>
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"x":[5000,5000],"y":[0,33900036.5],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,28378.900000000001],"y":[5000000,5000000],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[24483,24384,25517,23006,25341,25799,24513,23867],"y":[20637971,18809481,25348649,15163415,24448408,30818215,19126151,21164133],"text":["sample_id: SRR1039508<br />qc_status: QC pass","sample_id: SRR1039509<br />qc_status: QC pass","sample_id: SRR1039512<br />qc_status: QC pass","sample_id: SRR1039513<br />qc_status: QC pass","sample_id: SRR1039516<br />qc_status: QC pass","sample_id: SRR1039517<br />qc_status: QC pass","sample_id: SRR1039520<br />qc_status: QC pass","sample_id: SRR1039521<br />qc_status: QC pass"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"QC pass","legendgroup":"QC pass","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":41.778974318367787,"l":54.794520547945211},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,28378.900000000001],"tickmode":"array","ticktext":["0","10000","20000"],"tickvals":[0,10000,20000],"categoryorder":"array","categoryarray":["0","10000","20000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"Number of genes with non-zero counts","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,33900036.5],"tickmode":"array","ticktext":["0e+00","1e+07","2e+07","3e+07"],"tickvals":[0,10000000,20000000,30000000],"categoryorder":"array","categoryarray":["0e+00","1e+07","2e+07","3e+07"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"Total number of counts","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"QC status","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"687f5018a634":{"xintercept":{},"type":"scatter"},"687f6a6b8db8":{"yintercept":{}},"687f7d9d1fd3":{"x":{},"y":{},"label":{},"colour":{}},"687f32a2257a":{"x":{},"y":{},"label":{}}},"cur_data":"687f5018a634","visdat":{"687f5018a634":["function (y) ","x"],"687f6a6b8db8":["function (y) ","x"],"687f7d9d1fd3":["function (y) ","x"],"687f32a2257a":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="normalize">Normalize<a class="anchor" aria-label="anchor" href="#normalize"></a>
</h3>
<p>Here, we apply a normalization to the expression data, making samples
comparable by reducing variability due to technical differences. For
datasets with few samples, <code>rlog</code> is the preferred
normalization and when more samples are present, <code>vst</code> is
applied.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_gexp.html">normalize_gexp</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h2>
<p>Principal component analysis (PCA) identifies the major patterns in
the dataset. These patterns help explore similarities or differences
among samples based on gene expression.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">heatmap_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  DNA_repair <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BRCA1"</span>, <span class="st">"BRCA2"</span>, <span class="st">"ATM"</span>, <span class="st">"ATR"</span>, <span class="st">"CHEK1"</span>, <span class="st">"CHEK2"</span>, <span class="st">"RAD51"</span>, <span class="st">"XRCC1"</span><span class="op">)</span>,</span>
<span>  Metabolism <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACLY"</span>, <span class="st">"ACACA"</span>, <span class="st">"FASN"</span>, <span class="st">"G6PD"</span>, <span class="st">"HK2"</span>, <span class="st">"LDHA"</span>, <span class="st">"PDK1"</span><span class="op">)</span>,</span>
<span>  Cell_Cycle <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CDK1"</span>, <span class="st">"CDK2"</span>, <span class="st">"CCND1"</span>, <span class="st">"CCNE1"</span>, <span class="st">"RB1"</span>, <span class="st">"TP53"</span>, <span class="st">"E2F1"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">[[</span><span class="st">"pca_res"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pca_gexp.html">pca_gexp</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">annotation</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exp_cluster"</span>, <span class="st">"path_cluster"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">exp_data</span>, color <span class="op">=</span> <span class="va">annotation</span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"data":[{"x":[14.839069421975847,24.593807654692181,16.383181911026984,25.967675811227551],"y":[5.0172517058590902,9.0421838417274021,-30.394274516115658,11.425730228598246],"text":["sample_id: SRR1039509<br />dex: trt","sample_id: SRR1039513<br />dex: trt","sample_id: SRR1039517<br />dex: trt","sample_id: SRR1039521<br />dex: trt"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":7.559055118110237,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"trt","legendgroup":"trt","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-21.384638314040082,-19.659551572912438,-20.542436740837015,-20.197108171133014],"y":[10.241576532822902,9.2251012769538185,-31.202632473378614,16.645063403532838],"text":["sample_id: SRR1039508<br />dex: untrt","sample_id: SRR1039512<br />dex: untrt","sample_id: SRR1039516<br />dex: untrt","sample_id: SRR1039520<br />dex: untrt"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":7.559055118110237,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,191,196,1)"}},"hoveron":"points","name":"untrt","legendgroup":"untrt","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-23.752254020303464,28.335291517490933],"y":[0,0],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,0],"y":[-33.595017267224186,19.03744819737841],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":41.778974318367787,"l":43.105022831050235},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-23.752254020303464,28.335291517490933],"tickmode":"array","ticktext":["-20","-10","0","10","20"],"tickvals":[-20,-10,0,10.000000000000004,20.000000000000004],"categoryorder":"array","categoryarray":["-20","-10","0","10","20"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.6529680365296811,"tickwidth":0.33208800332088001,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(222,222,222,1)","gridwidth":0.33208800332088001,"zeroline":false,"anchor":"y","title":{"text":"PC 1 (24.6%)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-33.595017267224186,19.03744819737841],"tickmode":"array","ticktext":["-30","-20","-10","0","10"],"tickvals":[-30,-20,-10,0,10],"categoryorder":"array","categoryarray":["-30","-20","-10","0","10"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.6529680365296811,"tickwidth":0.33208800332088001,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(222,222,222,1)","gridwidth":0.33208800332088001,"zeroline":false,"anchor":"x","title":{"text":"PC 2 (18.6%)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(0,0,0,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"dex","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"687f66556218":{"label":{},"x":{},"y":{},"colour":{},"type":"scatter"},"687f6181fd9e":{"yintercept":{}},"687f133fbecf":{"xintercept":{}}},"cur_data":"687f66556218","visdat":{"687f66556218":["function (y) ","x"],"687f6181fd9e":["function (y) ","x"],"687f133fbecf":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="diffexp">Diffexp<a class="anchor" aria-label="anchor" href="#diffexp"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">dex</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">dex</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"untrt"</span>, <span class="st">"trt"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">diffexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffExpAnalysis.html">diffExpAnalysis</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,</span>
<span>                                          sampleInfo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span>,</span>
<span>                                          method <span class="op">=</span> <span class="va">diffexpMethod</span>, cutoff <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                          annotation <span class="op">=</span> <span class="va">annotation</span><span class="op">)</span></span></code></pre></div>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Diffexp Filtering</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="va">diffexpMethod</span><span class="op">)</span> <span class="op">==</span> <span class="st">"limma"</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">diffexp</span> <span class="op">&lt;-</span> <span class="va">diffexp</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span></span>
<span>      log2FoldChange <span class="op">=</span> <span class="va">logFC</span>,</span>
<span>      pvalue <span class="op">=</span> <span class="va">P.Value</span>,</span>
<span>      padj <span class="op">=</span> <span class="va">adj.P.Val</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">diffexp_filtered</span> <span class="op">&lt;-</span> <span class="va">diffexp</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">padj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Sélectionner les 30 gènes les plus différentiellement exprimés</span></span>
<span><span class="va">top_de_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">diffexp_filtered</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">16</span><span class="op">]</span></span></code></pre></div>
<p>##Volcano plot</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">volcano_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_exp_volcano.html">plot_exp_volcano</a></span><span class="op">(</span><span class="va">diffexp</span>, <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">volcano_plot</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 10107 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text_repel()`).</span></span></code></pre>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">volcano_plot</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 10107 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text_repel()`).</span></span></code></pre>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="pathway-analysis">Pathway analysis<a class="anchor" aria-label="anchor" href="#pathway-analysis"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_collections.html" class="external-link">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/scroll_box.html" class="external-link">scroll_box</a></span><span class="op">(</span>height <span class="op">=</span> <span class="st">"300px"</span><span class="op">)</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; ">
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_collection
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_subcollection
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_collection_name
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
num_genesets
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
C1
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Positional
</td>
<td style="text-align:right;">
302
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CGP
</td>
<td style="text-align:left;">
Chemical and Genetic Perturbations
</td>
<td style="text-align:right;">
3494
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP
</td>
<td style="text-align:left;">
Canonical Pathways
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:BIOCARTA
</td>
<td style="text-align:left;">
BioCarta Pathways
</td>
<td style="text-align:right;">
292
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:KEGG_LEGACY
</td>
<td style="text-align:left;">
KEGG Legacy Pathways
</td>
<td style="text-align:right;">
186
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:KEGG_MEDICUS
</td>
<td style="text-align:left;">
KEGG Medicus Pathways
</td>
<td style="text-align:right;">
658
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:PID
</td>
<td style="text-align:left;">
PID Pathways
</td>
<td style="text-align:right;">
196
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:REACTOME
</td>
<td style="text-align:left;">
Reactome Pathways
</td>
<td style="text-align:right;">
1736
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:WIKIPATHWAYS
</td>
<td style="text-align:left;">
WikiPathways
</td>
<td style="text-align:right;">
830
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
MIR:MIRDB
</td>
<td style="text-align:left;">
miRDB
</td>
<td style="text-align:right;">
2377
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
MIR:MIR_LEGACY
</td>
<td style="text-align:left;">
MIR_Legacy
</td>
<td style="text-align:right;">
221
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
TFT:GTRD
</td>
<td style="text-align:left;">
GTRD
</td>
<td style="text-align:right;">
505
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
TFT:TFT_LEGACY
</td>
<td style="text-align:left;">
TFT_Legacy
</td>
<td style="text-align:right;">
610
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
3CA
</td>
<td style="text-align:left;">
Curated Cancer Cell Atlas gene sets
</td>
<td style="text-align:right;">
148
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
CGN
</td>
<td style="text-align:left;">
Cancer Gene Neighborhoods
</td>
<td style="text-align:right;">
427
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
CM
</td>
<td style="text-align:left;">
Cancer Modules
</td>
<td style="text-align:right;">
431
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:BP" class="uri">GO:BP</a>
</td>
<td style="text-align:left;">
GO Biological Process
</td>
<td style="text-align:right;">
7608
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:CC" class="uri">GO:CC</a>
</td>
<td style="text-align:left;">
GO Cellular Component
</td>
<td style="text-align:right;">
1026
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:MF" class="uri">GO:MF</a>
</td>
<td style="text-align:left;">
GO Molecular Function
</td>
<td style="text-align:right;">
1820
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
HPO
</td>
<td style="text-align:left;">
Human Phenotype Ontology
</td>
<td style="text-align:right;">
5653
</td>
</tr>
<tr>
<td style="text-align:left;">
C6
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Oncogenic Signature
</td>
<td style="text-align:right;">
189
</td>
</tr>
<tr>
<td style="text-align:left;">
C7
</td>
<td style="text-align:left;">
IMMUNESIGDB
</td>
<td style="text-align:left;">
ImmuneSigDB
</td>
<td style="text-align:right;">
4872
</td>
</tr>
<tr>
<td style="text-align:left;">
C7
</td>
<td style="text-align:left;">
VAX
</td>
<td style="text-align:left;">
HIPC Vaccine Response
</td>
<td style="text-align:right;">
347
</td>
</tr>
<tr>
<td style="text-align:left;">
C8
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Cell Type Signature
</td>
<td style="text-align:right;">
840
</td>
</tr>
<tr>
<td style="text-align:left;">
H
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Hallmark
</td>
<td style="text-align:right;">
50
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_annotation_collection.html">get_annotation_collection</a></span><span class="op">(</span><span class="va">collections</span>, </span>
<span>                                      species <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pathwayResult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathwayAnalysis.html">pathwayAnalysis</a></span><span class="op">(</span><span class="va">diffexp_filtered</span>,</span>
<span>                              pathways <span class="op">=</span> <span class="va">pathways</span>,</span>
<span>                              method <span class="op">=</span> <span class="va">pathwayMethod</span>, species <span class="op">=</span> <span class="va">species</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pathwayResult</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pathwayResult</span><span class="op">$</span><span class="va">pathway</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">[[</span><span class="st">"pathwayEnrichment"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">pathwayResult</span></span>
<span></span>
<span><span class="co"># Trier selon PValue croissante (plus significatif en haut)</span></span>
<span><span class="va">pathwayResult_sorted</span> <span class="op">&lt;-</span> <span class="va">pathwayResult</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">pathwayResult</span><span class="op">$</span><span class="va">PValue</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co"># Garder uniquement les 10 premiers</span></span>
<span><span class="va">top10_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathwayResult_sorted</span>, <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathwayDotplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_pathway_dotplot.html">plot_pathway_dotplot</a></span><span class="op">(</span><span class="va">exp_data</span>, score_name <span class="op">=</span> <span class="st">"pathwayEnrichment"</span><span class="op">)</span></span>
<span><span class="va">pathwayDotplot</span></span></code></pre></div>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-12-1.png" width="700">
“SRR1039508” “SRR1039509” “SRR1039512” “SRR1039513” “SRR1039516” [6]
“SRR1039517” “SRR1039520” “SRR1039521”</p>
<p>On the heatmaps, we can tell that the top_genes expressions are
higher for the treated samples? It means that these genes are the ones
that are involved in the treatment response.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">heatmap_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"STOM"</span>,<span class="st">"PHC2"</span>, <span class="st">"MT2A"</span>, <span class="st">"ABAT"</span>,<span class="st">"DUSP1"</span>,<span class="st">"PER1"</span>,<span class="st">"KCTD12"</span>, <span class="st">"SAMHD1"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FAM171B"</span>,<span class="st">"MAOA"</span>,<span class="st">"NEXN"</span>,<span class="st">"DNAJB4"</span>,<span class="st">"ARHGEF2"</span>,<span class="st">"FKBP5"</span>,<span class="st">"JADE1"</span>,<span class="st">"RAP2B"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">heatmap_genes</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span><span class="op">)</span>, <span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_exp_heatmap.html">plot_exp_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, genes <span class="op">=</span> <span class="va">genes</span>, </span>
<span>                   annotations <span class="op">=</span> <span class="va">annotation</span>,</span>
<span>                   show_rownames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">100</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>                   hm_color_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>                   show_dend_row <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/sup/targeted/{pathwayMethod}/hm_genes_{i}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">hms</span>, ncol <span class="op">=</span> <span class="fl">2</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"results"</span>, <span class="st">"sup"</span>, <span class="st">"targeted"</span>, <span class="va">pathwayMethod</span>, <span class="st">"boxplots"</span><span class="op">)</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">pathway_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/score_pathways.html">score_pathways</a></span><span class="op">(</span><span class="va">exp_data</span>, <span class="va">pathways</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">[[</span><span class="st">"pathway_scores"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">pathway_scores</span></span>
<span></span>
<span><span class="va">boxplots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">heatmap_pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annotation</span>, <span class="kw">function</span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_path_boxplot.html">plot_path_boxplot</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                             pathway <span class="op">=</span> <span class="va">path</span>,</span>
<span>                   annotation <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                  color_var <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                   pt_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/sup/targeted/{pathwayMethod}/boxplots/box_{path}_{annotation}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu">flatten</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">boxplots</span>, nrows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">boxplots</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="sup_rnaseq_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Clarice Groeneveld, Chloe Witz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
