<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Unsupervised RNA-seq analysis • CAIBIrnaseq</title>
<!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Unsupervised RNA-seq analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CAIBIrnaseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation Guide</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">References</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-examples" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-examples">
<li><a class="dropdown-item" href="../articles/unsup_rnaseq.html">Unsupervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/sup_rnaseq.html">Supervised analysis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="unsup_rnaseq_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="unsup_rnaseq_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="unsup_rnaseq_files/plotly-binding-4.10.4/plotly.js"></script><script src="unsup_rnaseq_files/typedarray-0.1/typedarray.min.js"></script><link href="unsup_rnaseq_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="unsup_rnaseq_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="unsup_rnaseq_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="unsup_rnaseq_files/plotly-main-2.11.1/plotly-latest.min.js"></script><script src="unsup_rnaseq_files/kePrint-0.0.1/kePrint.js"></script><link href="unsup_rnaseq_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Unsupervised RNA-seq analysis</h1>
                        <h4 data-toc-skip class="author">Cordeliers
Artificial Intelligence and Bioinformatics</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/crcordeliers/CAIBIrnaseq.git/blob/HEAD/vignettes/unsup_rnaseq.Rmd" class="external-link"><code>vignettes/unsup_rnaseq.Rmd</code></a></small>
      <div class="d-none name"><code>unsup_rnaseq.Rmd</code></div>
    </div>

    
    
<p>For the 2 examples, we are using an open dataset which is given by
the <code>airway</code> package. It provides a gene expression dataset
derived from <strong>human bronchial epithelial cells</strong>, treated
or not with <strong>dexamethasone</strong> (a corticosteroid).</p>
<p>Here is an example of how the <strong>unsupervised</strong> part of
the <code>CAIBIrnaseq</code> package can be use.</p>
<p>First, we need to import the required packages :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
<p>Before analysing the dataset, we define the variables with the names
of the genes / pathways … that we want to analyse.</p>
<p>These variables will change in function of how your dataset is build
and which type of data it is.</p>
<p>If you find the definition of the variables heavy in your code, you
can create a <code>.yml</code> file, for example named
<code>params.yml</code>, call it in a cell, and it will automatically
import the variables. Your code will be clearer.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">species</span> <span class="op">&lt;-</span> <span class="st">"Homo sapiens"</span></span>
<span></span>
<span><span class="co"># Annotation variable to visualize</span></span>
<span><span class="va">plot_annotations</span> <span class="op">&lt;-</span> <span class="st">"dex"</span>  <span class="co"># treated or untreated</span></span>
<span></span>
<span><span class="co"># Quality parameters</span></span>
<span><span class="va">qc_min_nsamp</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">qc_min_gene_counts</span> <span class="op">&lt;-</span> <span class="fl">10</span>  </span>
<span></span>
<span><span class="co"># Clustering of expression</span></span>
<span><span class="va">exp_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>k <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Clustering of metadata</span></span>
<span><span class="va">metadata_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  pathway_scores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>k <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  microenv_scores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Pathway collections </span></span>
<span><span class="va">pathway_collections</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CP:REACTOME"</span>, <span class="st">"H"</span><span class="op">)</span>  <span class="co"># You can add as many pathways as you want it will only create more plots</span></span>
<span></span>
<span><span class="co"># Interesting genes</span></span>
<span><span class="va">heatmap_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NR3C1"</span>, <span class="st">"FKBP5"</span>, <span class="st">"TSC22D3"</span>, <span class="st">"ZBTB16"</span>, <span class="st">"PER1"</span><span class="op">)</span>,  <span class="co"># gènes régulés par le glucocorticoïde</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CYP1B1"</span>, <span class="st">"G6PD"</span>, <span class="st">"HMOX1"</span>, <span class="st">"NQO1"</span>, <span class="st">"SOD2"</span><span class="op">)</span>       <span class="co"># stress oxydatif / métabolisme</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Interesting pathways (adapt to the results)</span></span>
<span><span class="va">heatmap_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"REACTOME_INTERLEUKIN_36_PATHWAY"</span>,</span>
<span>  <span class="st">"REACTOME_IRS_ACTIVATION"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Genes for the boxplots</span></span>
<span><span class="va">boxplot_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FKBP5"</span>, <span class="st">"TSC22D3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pathways for the boxplots</span></span>
<span><span class="va">boxplot_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"REACTOME_INTERLEUKIN_36_PATHWAY"</span>,</span>
<span>  <span class="st">"REACTOME_IRS_ACTIVATION"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Corrélations entre gènes</span></span>
<span><span class="va">correlation_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FKBP5"</span>, <span class="st">"TSC22D3"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FKBP5"</span>, <span class="st">"ZBTB16"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pathways correlation</span></span>
<span><span class="va">correlation_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"REACTOME_INTERLEUKIN_36_PATHWAY"</span>, <span class="st">"REACTOME_IRS_ACTIVATION"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"REACTOME_INTERLEUKIN_36_PATHWAY"</span>, <span class="st">"REACTOME_UBIQUINOL_BIOSYNTHESIS"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Do not forget to import the package ! ;-)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://crcordeliers.github.io/CAIBIrnaseq">CAIBIrnaseq</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>This section loads the RNA-seq dataset for analysis. It ensures the
correct input file is used, as specified in the parameters.
rebase_gexp</p>
<p>Ensure your dataset is in a <code>Summarized Experiment</code>
object, because all the used functions below works with
SummarizedExperiment input.</p>
<p>If you want to know more about this type of object, please click
here: <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">Bioconductor</a></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span>, package<span class="op">=</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="va">airway</span></span></code></pre></div>
<p>Next cell is for the <code>airway dataset</code>. The goal is to
match the names of the variables used in the package with the names of
the dataset.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Huber-group-EMBL/biomaRt" class="external-link">biomaRt</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_length_kb</span> <span class="op">&lt;-</span> </span>
<span>  <span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_seq_end</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_seq_start</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1000</span></span>
<span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ensembl"</span>, dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">gene_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span></span>
<span></span>
<span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_gene_id"</span>, <span class="st">"description"</span><span class="op">)</span>,</span>
<span>               filters <span class="op">=</span> <span class="st">"ensembl_gene_id"</span>,</span>
<span>               values <span class="op">=</span> <span class="va">gene_ids</span>,</span>
<span>               mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span></span>
<span></span>
<span><span class="va">matched</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">annot</span><span class="op">$</span><span class="va">ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">gene_description</span> <span class="op">&lt;-</span> <span class="va">annot</span><span class="op">$</span><span class="va">description</span><span class="op">[</span><span class="va">matched</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h2>
<p>Most datasets use ensembl gene ID by default after alignment, so this
step rebases the expression data to gene names. This ensures consistency
in naming for downstream analyses.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rebase_gexp.html">rebase_gexp</a></span><span class="op">(</span><span class="va">exp_data</span>, annotation <span class="op">=</span> <span class="st">"gene_name"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="filter">Filter<a class="anchor" aria-label="anchor" href="#filter"></a>
</h3>
<p>Here, we filter out genes expressed in too few samples or with very
low counts. This removes noise from the data and focuses on meaningful
gene expressions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_gexp.html">filter_gexp</a></span><span class="op">(</span><span class="va">exp_data</span>,</span>
<span>                        min_nsamp <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                        min_counts <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Visualization of the filtering process to ensure the criteria applied
align with the dataset’s characteristics:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">sample_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_qc_filters.html">plot_qc_filters</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 8 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_text_repel()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning in geom2trace.default(dots[[1L]][[1L]], dots[[2L]][[1L]], dots[[3L]][[1L]]): geom_GeomTextRepel() has yet to be implemented in plotly.</span></span>
<span><span class="co">##   If you'd like to see this geom implemented,</span></span>
<span><span class="co">##   Please open an issue with your example code at</span></span>
<span><span class="co">##   https://github.com/ropensci/plotly/issues</span></span></code></pre>
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"x":[5000,5000],"y":[0,33900036.5],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,28378.900000000001],"y":[5000000,5000000],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[24483,24384,25517,23006,25341,25799,24513,23867],"y":[20637971,18809481,25348649,15163415,24448408,30818215,19126151,21164133],"text":["sample_id: SRR1039508<br />qc_status: QC pass","sample_id: SRR1039509<br />qc_status: QC pass","sample_id: SRR1039512<br />qc_status: QC pass","sample_id: SRR1039513<br />qc_status: QC pass","sample_id: SRR1039516<br />qc_status: QC pass","sample_id: SRR1039517<br />qc_status: QC pass","sample_id: SRR1039520<br />qc_status: QC pass","sample_id: SRR1039521<br />qc_status: QC pass"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)"}},"hoveron":"points","name":"QC pass","legendgroup":"QC pass","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":41.778974318367787,"l":54.794520547945211},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,28378.900000000001],"tickmode":"array","ticktext":["0","10000","20000"],"tickvals":[0,10000,20000],"categoryorder":"array","categoryarray":["0","10000","20000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"Number of genes with non-zero counts","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,33900036.5],"tickmode":"array","ticktext":["0e+00","1e+07","2e+07","3e+07"],"tickvals":[0,10000000,20000000,30000000],"categoryorder":"array","categoryarray":["0e+00","1e+07","2e+07","3e+07"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"Total number of counts","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"QC status","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"2ae2440109f2":{"xintercept":{},"type":"scatter"},"2ae24a85b371":{"yintercept":{}},"2ae27809ebd6":{"x":{},"y":{},"label":{},"colour":{}},"2ae22ce38fc":{"x":{},"y":{},"label":{}}},"cur_data":"2ae2440109f2","visdat":{"2ae2440109f2":["function (y) ","x"],"2ae24a85b371":["function (y) ","x"],"2ae27809ebd6":["function (y) ","x"],"2ae22ce38fc":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="normalize">Normalize<a class="anchor" aria-label="anchor" href="#normalize"></a>
</h3>
<p>Here, we apply a normalization to the expression data, making samples
comparable by reducing variability due to technical differences. For
datasets with few samples, <code>rlog</code> is the preferred
normalization and when more samples are present, <code>vst</code> is
applied.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_gexp.html">normalize_gexp</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h2>
<p>Principal component analysis (PCA) identifies the major patterns in
the dataset. These patterns help explore similarities or differences
among samples based on gene expression.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca_res</span> <span class="op">=</span> <span class="fu"><a href="../reference/pca_gexp.html">pca_gexp</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span><span class="va">exp_data</span><span class="op">@</span><span class="va">metadata</span><span class="op">[[</span><span class="st">"pca_res"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">pca_res</span></span>
<span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">plot_annotations</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exp_cluster"</span>, <span class="st">"path_cluster"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="va">exp_data</span>, color <span class="op">=</span> <span class="va">plot_annotations</span><span class="op">)</span></span></code></pre></div>
<div class="plotly html-widget html-fill-item" id="htmlwidget-e5c8c404fe174e4c81bd" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-e5c8c404fe174e4c81bd">{"x":{"data":[{"x":[14.839069421975847,24.593807654692181,16.383181911026984,25.967675811227551],"y":[5.0172517058590902,9.0421838417274021,-30.394274516115658,11.425730228598246],"text":["sample_id: SRR1039509<br />dex: trt","sample_id: SRR1039513<br />dex: trt","sample_id: SRR1039517<br />dex: trt","sample_id: SRR1039521<br />dex: trt"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":7.559055118110237,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"trt","legendgroup":"trt","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-21.384638314040082,-19.659551572912438,-20.542436740837015,-20.197108171133014],"y":[10.241576532822902,9.2251012769538185,-31.202632473378614,16.645063403532838],"text":["sample_id: SRR1039508<br />dex: untrt","sample_id: SRR1039512<br />dex: untrt","sample_id: SRR1039516<br />dex: untrt","sample_id: SRR1039520<br />dex: untrt"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":7.559055118110237,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,191,196,1)"}},"hoveron":"points","name":"untrt","legendgroup":"untrt","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-23.752254020303464,28.335291517490933],"y":[0,0],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,0],"y":[-33.595017267224186,19.03744819737841],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":27.824636418824401,"r":7.3059360730593621,"b":41.778974318367787,"l":43.105022831050235},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-23.752254020303464,28.335291517490933],"tickmode":"array","ticktext":["-20","-10","0","10","20"],"tickvals":[-20,-10,0,10.000000000000004,20.000000000000004],"categoryorder":"array","categoryarray":["-20","-10","0","10","20"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.6529680365296811,"tickwidth":0.33208800332088001,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(222,222,222,1)","gridwidth":0.33208800332088001,"zeroline":false,"anchor":"y","title":{"text":"PC 1 (24.6%)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-33.595017267224186,19.03744819737841],"tickmode":"array","ticktext":["-30","-20","-10","0","10"],"tickvals":[-30,-20,-10,0,10],"categoryorder":"array","categoryarray":["-30","-20","-10","0","10"],"nticks":null,"ticks":"outside","tickcolor":"rgba(0,0,0,1)","ticklen":3.6529680365296811,"tickwidth":0.33208800332088001,"showticklabels":true,"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(222,222,222,1)","gridwidth":0.33208800332088001,"zeroline":false,"anchor":"x","title":{"text":"PC 2 (18.6%)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(0,0,0,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"dex","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"2ae21429feff":{"label":{},"x":{},"y":{},"colour":{},"type":"scatter"},"2ae26c70bc72":{"yintercept":{}},"2ae22075317f":{"xintercept":{}}},"cur_data":"2ae21429feff","visdat":{"2ae21429feff":["function (y) ","x"],"2ae26c70bc72":["function (y) ","x"],"2ae22075317f":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p>If you want something more visual, you can add a circular/oval shape
to circle the different genotypes of samples, use the
<code>fviz_pca_ind</code> function from the <code>factoextra</code>
package. With this dataset, it is not relevant but it can be with your
persona dataset. Here, it highlights the fact that the 2 groups are not
crossing each other. The <code>trt</code> group has more PC1, whereas
the <code>untrt</code> group has less. We could conclude that PC1 is
more represented in the treated samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.sthda.com/english/rpkgs/factoextra" class="external-link">factoextra</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<pre><code><span><span class="co">## Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">$</span><span class="va">dex</span>  <span class="co"># here we want to split in function of `treated` and `untreated`</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/factoextra/man/fviz_pca.html" class="external-link">fviz_pca_ind</a></span><span class="op">(</span><span class="va">pca_res</span>,</span>
<span>             geom <span class="op">=</span> <span class="st">"point"</span>,</span>
<span>             habillage <span class="op">=</span> <span class="va">groups</span>,</span>
<span>             palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#00AFBB"</span>, <span class="st">"#E7B800"</span><span class="op">)</span>,  <span class="co"># Personalized colors</span></span>
<span>             addEllipses <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>             ellipse.type <span class="op">=</span> <span class="st">"confidence"</span>,</span>
<span>             repel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>             label <span class="op">=</span> <span class="st">"none"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="unsupervised-clustering">Unsupervised clustering<a class="anchor" aria-label="anchor" href="#unsupervised-clustering"></a>
</h2>
<p>Here, we group samples based on expression patterns without prior
knowledge using hierarchical clustering on either a selected gene list
from the parameters or, by default, the 2000 most highly expressed
genes.</p>
<p>This can be useful for discovering sample subgroups or new biological
insights.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_exp.html">cluster_exp</a></span><span class="op">(</span><span class="va">exp_data</span>, k <span class="op">=</span> <span class="va">exp_cluster</span><span class="op">$</span><span class="va">k</span>, genes <span class="op">=</span> <span class="va">exp_cluster</span><span class="op">$</span><span class="va">genes</span>, n_pcs <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p>Visual representation of expression levels for HVG across clusters,
highlighting distinct patterns.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hvg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/highly_variable_genes.html">highly_variable_genes</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span><span class="va">exp_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>k <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_exp_heatmap.html">plot_exp_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>,  genes <span class="op">=</span> <span class="va">hvg</span>, </span>
<span>                 annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">plot_annotations</span>, <span class="st">"exp_cluster"</span><span class="op">)</span>,</span>
<span>                 show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                 hm_color_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>                 fname <span class="op">=</span> <span class="st">"results/clustering/heatmap_2000hvg_exp_cluster.pdf"</span><span class="op">)</span></span>
<span><span class="va">hm</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot%20hvg-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="pathway-activity">Pathway activity<a class="anchor" aria-label="anchor" href="#pathway-activity"></a>
</h2>
<p>Pathway analysis enables us to understand the functional implications
of gene expression changes. Here, we analyze the dataset for pathway
activity using two methods.</p>
<div class="section level3">
<h3 id="progeny">PROGENy<a class="anchor" aria-label="anchor" href="#progeny"></a>
</h3>
<p>PROGENy is a collection of only 14 core pathway responsive genes from
large signaling perturbation experiments. For more information see the
<a href="https://www.nature.com/articles/s41467-017-02391-6" class="external-link">original
paper</a>.</p>
<p>The returned plot will give us information about the pathways that
are activated for each sample. There is especially one pathway that is
highly activated : <code>EGFR</code> , in the sample
<code>SRR1039517</code></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">progeny_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/score_progeny.html">score_progeny</a></span><span class="op">(</span><span class="va">exp_data</span>, species <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span>         </span>
<span><span class="va">tmp</span><span class="op">[[</span><span class="st">"progeny_scores"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">progeny_scores</span>  </span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tmp</span>  </span>
<span></span>
<span><span class="fu"><a href="../reference/plot_progeny_heatmap.html">plot_progeny_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, annotations <span class="op">=</span> <span class="va">plot_annotations</span>,</span>
<span>                     fname <span class="op">=</span> <span class="st">"results/pathways/hm_progeny_scores.pdf"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prep_scores_hm(exp_data, progeny_scores): 'sample_id' already exists</span></span>
<span><span class="co">## in colData and will be overwritten.</span></span></code></pre>
<p><img src="unsup_rnaseq_files/figure-html/progenyprovided-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">progeny_scores</span>, file <span class="op">=</span> <span class="st">"results/pathways/progeny_scores.csv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathways">Pathways<a class="anchor" aria-label="anchor" href="#pathways"></a>
</h3>
<p>Pathway collections available in the MSIGdb can be specified in the
parameters. These pathways are scored and ranked by their variance in
the data. These are the available collections (use
<code>gs_subcollection</code> as name except for Hallmarks, which should
be ‘H’).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">kableExtra</a></span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr_collections.html" class="external-link">msigdbr_collections</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/scroll_box.html" class="external-link">scroll_box</a></span><span class="op">(</span>height <span class="op">=</span> <span class="st">"300px"</span><span class="op">)</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:300px; ">
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_collection
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_subcollection
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gs_collection_name
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
num_genesets
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
C1
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Positional
</td>
<td style="text-align:right;">
302
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CGP
</td>
<td style="text-align:left;">
Chemical and Genetic Perturbations
</td>
<td style="text-align:right;">
3494
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP
</td>
<td style="text-align:left;">
Canonical Pathways
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:BIOCARTA
</td>
<td style="text-align:left;">
BioCarta Pathways
</td>
<td style="text-align:right;">
292
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:KEGG_LEGACY
</td>
<td style="text-align:left;">
KEGG Legacy Pathways
</td>
<td style="text-align:right;">
186
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:KEGG_MEDICUS
</td>
<td style="text-align:left;">
KEGG Medicus Pathways
</td>
<td style="text-align:right;">
658
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:PID
</td>
<td style="text-align:left;">
PID Pathways
</td>
<td style="text-align:right;">
196
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:REACTOME
</td>
<td style="text-align:left;">
Reactome Pathways
</td>
<td style="text-align:right;">
1736
</td>
</tr>
<tr>
<td style="text-align:left;">
C2
</td>
<td style="text-align:left;">
CP:WIKIPATHWAYS
</td>
<td style="text-align:left;">
WikiPathways
</td>
<td style="text-align:right;">
830
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
MIR:MIRDB
</td>
<td style="text-align:left;">
miRDB
</td>
<td style="text-align:right;">
2377
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
MIR:MIR_LEGACY
</td>
<td style="text-align:left;">
MIR_Legacy
</td>
<td style="text-align:right;">
221
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
TFT:GTRD
</td>
<td style="text-align:left;">
GTRD
</td>
<td style="text-align:right;">
505
</td>
</tr>
<tr>
<td style="text-align:left;">
C3
</td>
<td style="text-align:left;">
TFT:TFT_LEGACY
</td>
<td style="text-align:left;">
TFT_Legacy
</td>
<td style="text-align:right;">
610
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
3CA
</td>
<td style="text-align:left;">
Curated Cancer Cell Atlas gene sets
</td>
<td style="text-align:right;">
148
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
CGN
</td>
<td style="text-align:left;">
Cancer Gene Neighborhoods
</td>
<td style="text-align:right;">
427
</td>
</tr>
<tr>
<td style="text-align:left;">
C4
</td>
<td style="text-align:left;">
CM
</td>
<td style="text-align:left;">
Cancer Modules
</td>
<td style="text-align:right;">
431
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:BP" class="uri">GO:BP</a>
</td>
<td style="text-align:left;">
GO Biological Process
</td>
<td style="text-align:right;">
7608
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:CC" class="uri">GO:CC</a>
</td>
<td style="text-align:left;">
GO Cellular Component
</td>
<td style="text-align:right;">
1026
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
<a href="GO:MF" class="uri">GO:MF</a>
</td>
<td style="text-align:left;">
GO Molecular Function
</td>
<td style="text-align:right;">
1820
</td>
</tr>
<tr>
<td style="text-align:left;">
C5
</td>
<td style="text-align:left;">
HPO
</td>
<td style="text-align:left;">
Human Phenotype Ontology
</td>
<td style="text-align:right;">
5653
</td>
</tr>
<tr>
<td style="text-align:left;">
C6
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Oncogenic Signature
</td>
<td style="text-align:right;">
189
</td>
</tr>
<tr>
<td style="text-align:left;">
C7
</td>
<td style="text-align:left;">
IMMUNESIGDB
</td>
<td style="text-align:left;">
ImmuneSigDB
</td>
<td style="text-align:right;">
4872
</td>
</tr>
<tr>
<td style="text-align:left;">
C7
</td>
<td style="text-align:left;">
VAX
</td>
<td style="text-align:left;">
HIPC Vaccine Response
</td>
<td style="text-align:right;">
347
</td>
</tr>
<tr>
<td style="text-align:left;">
C8
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Cell Type Signature
</td>
<td style="text-align:right;">
840
</td>
</tr>
<tr>
<td style="text-align:left;">
H
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Hallmark
</td>
<td style="text-align:right;">
50
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_annotation_collection.html">get_annotation_collection</a></span><span class="op">(</span><span class="va">pathway_collections</span>, </span>
<span>                                      species <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in FUN(X[[i]], ...): Collection `H` not found in MSigDB. Skipping.</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/score_pathways.html">score_pathways</a></span><span class="op">(</span><span class="va">exp_data</span>, <span class="va">pathways</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">[[</span><span class="st">"pathway_scores"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">pathway_scores</span></span>
<span></span>
<span><span class="va">collections</span> <span class="op">&lt;-</span> <span class="va">pathway_collections</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span>collapse <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove</a></span><span class="op">(</span><span class="st">"\\:"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_pathway_heatmap.html">plot_pathway_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, annotations <span class="op">=</span> <span class="va">plot_annotations</span>,</span>
<span>                    fwidth <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                    fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span></span>
<span>                      <span class="st">"results/pathways/hm_paths_{collections}_top20.pdf"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prep_scores_hm(exp_data, pathway_scores, pathways): 'sample_id'</span></span>
<span><span class="co">## already exists in colData and will be overwritten.</span></span></code></pre>
<p><img src="unsup_rnaseq_files/figure-html/pathways-1.png" width="960"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">pathways</span>, file <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/pathways/paths_{collections}.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="microenvironment-scores">Microenvironment scores<a class="anchor" aria-label="anchor" href="#microenvironment-scores"></a>
</h2>
<p>This step calculates immune and stromal cell type abundances using
MCPcounter or mMCPcounter. It helps to infer the composition of the
tumor microenvironment or similar contexts.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mcp_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mcp_counter.html">mcp_counter</a></span><span class="op">(</span><span class="va">exp_data</span>, species <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span><span class="op">[[</span><span class="st">"microenv_scores"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mcp_scores</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_microenv_heatmap.html">plot_microenv_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dex"</span>, <span class="st">"exp_cluster"</span><span class="op">)</span>,</span>
<span>                      fname <span class="op">=</span> <span class="st">"results/tme/heatSmap_mcpcounter.pdf"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prep_scores_hm(exp_data, microenv_scores): 'sample_id' already</span></span>
<span><span class="co">## exists in colData and will be overwritten.</span></span></code></pre>
<p><img src="unsup_rnaseq_files/figure-html/microenvironment-1.png" width="700"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">mcp_scores</span>, file <span class="op">=</span> <span class="st">"results/tme/scores_mcpcounter.csv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="targeted-plots">Targeted plots<a class="anchor" aria-label="anchor" href="#targeted-plots"></a>
</h2>
<p>This section focuses on visualizing specific genes or pathways of
interest, as specified in the parameters.</p>
<div class="section level3">
<h3 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h3>
<p>Generates heatmaps for pre-selected genes of interest to observe
their expression across samples or conditions.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">gene_annot</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">exp_data</span><span class="op">)</span></span>
<span>  <span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">heatmap_genes</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span><span class="op">)</span>, <span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">heatmap_genes</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="../reference/plot_exp_heatmap.html">plot_exp_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, genes <span class="op">=</span> <span class="va">genes</span>, </span>
<span>                   annotations <span class="op">=</span> <span class="va">plot_annotations</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/targeted/hm_genes_{i}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">hms</span>, ncol <span class="op">=</span> <span class="fl">2</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot_hm_genes-1.png" width="960"></p>
<div class="section level4">
<h4 id="selected-pathways">Selected pathways<a class="anchor" aria-label="anchor" href="#selected-pathways"></a>
</h4>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">valid_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">heatmap_pathways</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pathway_scores</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_pathway_heatmap.html">plot_pathway_heatmap</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                     annotations <span class="op">=</span> <span class="va">plot_annotations</span>, </span>
<span>                     pathways <span class="op">=</span> <span class="va">valid_pathways</span>,</span>
<span>                     fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/targeted/hm_pathways_selected.pdf"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prep_scores_hm(exp_data, pathway_scores, pathways): 'sample_id'</span></span>
<span><span class="co">## already exists in colData and will be overwritten.</span></span></code></pre>
<p><img src="unsup_rnaseq_files/figure-html/plot_hm_pathways-1.png" width="672"></p>
</div>
</div>
<div class="section level3">
<h3 id="boxplots">Boxplots<a class="anchor" aria-label="anchor" href="#boxplots"></a>
</h3>
<p>Boxplots provide a clear comparison of expression levels across
experimental groups or conditions. All jobs still running at 10:00 on
Friday will be killed as part of the maintenance. #### Selected
genes</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">boxplot_genes</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="va">plot_annotations</span></span>
<span></span>
<span><span class="va">boxplots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">genes</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_exp_boxplot.html">plot_exp_boxplot</a></span><span class="op">(</span><span class="va">exp_data</span>, gene <span class="op">=</span> <span class="va">gene</span>, </span>
<span>                   annotation <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                   color_var <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                   pt_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/targeted/boxplots/box_{gene}_{annotation}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu">flatten</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">boxplots</span>, nrows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">boxplots</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot_box_genes2-1.png" width="960"></p>
<div class="section level4">
<h4 id="selected-pathways-1">Selected pathways<a class="anchor" aria-label="anchor" href="#selected-pathways-1"></a>
</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="va">boxplot_pathways</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="va">plot_annotations</span></span>
<span></span>
<span><span class="va">boxplots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">plt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_path_boxplot.html">plot_path_boxplot</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                             pathway <span class="op">=</span> <span class="va">path</span>,</span>
<span>                   annotation <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                   color_var <span class="op">=</span> <span class="va">annotation</span>, </span>
<span>                   pt_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/targeted/boxplots/box_{path}_{annotation}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu">flatten</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">boxplots</span>, nrows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">boxplots</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot_box_paths2-1.png" width="960"></p>
</div>
</div>
<div class="section level3">
<h3 id="correlations">Correlations<a class="anchor" aria-label="anchor" href="#correlations"></a>
</h3>
<p>This section visualizes relationships between pairs of genes or
pathways by plotting their expression/activity correlations. Correlation
analysis can reveal important co-regulation or interaction patterns,
helping to uncover biologically meaningful relationships.</p>
<div class="section level4">
<h4 id="selected-genes">Selected genes<a class="anchor" aria-label="anchor" href="#selected-genes"></a>
</h4>
<p>Here we plot the correlation between selected gene pairs across the
dataset. Each pair is plotted separately, and color-coded by sample
annotation.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pairs</span> <span class="op">&lt;-</span> <span class="va">correlation_genes</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="va">plot_annotations</span></span>
<span></span>
<span><span class="va">cor_plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">gene_pairs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">gene_pair</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="../reference/plot_exp_scatter.html">plot_exp_scatter</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                   gene1 <span class="op">=</span> <span class="va">gene_pair</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                   gene2 <span class="op">=</span> <span class="va">gene_pair</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>                   color_var <span class="op">=</span> <span class="va">annot</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span></span>
<span>                     <span class="st">"results/targeted/correlations/cor_{gene_pair[1]}_{gene_pair[2]}_color={annot}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu">flatten</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">cor_plts</span>, nrows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cor_plts</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot_correlation_genes2-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="selected-pathways-2">Selected pathways<a class="anchor" aria-label="anchor" href="#selected-pathways-2"></a>
</h4>
<p>Correlation plots for selected pathways can help identify
similarities or differences in pathway activity patterns across samples.
Each pathway pair is plotted separately and color-coded by sample
annotation to illustrate trends within each condition.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_pairs</span> <span class="op">&lt;-</span><span class="va">correlation_pathways</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="va">plot_annotations</span></span>
<span></span>
<span><span class="va">cor_plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">path_pairs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">path_pair</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="kw">function</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="../reference/plot_path_scatter.html">plot_path_scatter</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                   pathway1 <span class="op">=</span> <span class="va">path_pair</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                   pathway2 <span class="op">=</span> <span class="va">path_pair</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,<span class="co">## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was </span></span>
<span>                   color_var <span class="op">=</span> <span class="va">annot</span>,</span>
<span>                   fname <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span></span>
<span>                     <span class="st">"results/targeted/correlations/cor_{path_pair[1]}_{path_pair[2]}_color={annot}.pdf"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu">flatten</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">cor_plts</span>, nrows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cor_plts</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>, guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="unsup_rnaseq_files/figure-html/plot_correlation_paths2-1.png" width="960"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="cluster-using-metadata">Cluster using metadata<a class="anchor" aria-label="anchor" href="#cluster-using-metadata"></a>
</h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">metadata_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">type</span> <span class="kw">in</span> <span class="va">types</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">exp_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_metadata.html">cluster_metadata</a></span><span class="op">(</span><span class="va">exp_data</span>, </span>
<span>                   metadata_name <span class="op">=</span> <span class="va">type</span>, </span>
<span>                   k <span class="op">=</span> <span class="va">metadata_clusters</span><span class="op">[[</span><span class="va">type</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">k</span>, </span>
<span>                   features <span class="op">=</span> <span class="va">metadata_clusters</span><span class="op">[[</span><span class="va">type</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">features</span>,</span>
<span>                   n_pcs <span class="op">=</span> <span class="fl">3</span> <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="save-summarizedexperiment">Save SummarizedExperiment<a class="anchor" aria-label="anchor" href="#save-summarizedexperiment"></a>
</h2>
<p>The final step saves the processed dataset and results. This ensures
all outputs can be revisited or shared for further analysis.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">exp_data</span>, file <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_glue.html" class="external-link">str_glue</a></span><span class="op">(</span><span class="st">"results/data_SummarizedExp_{lubridate::today()}.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="report-parameters">Report parameters<a class="anchor" aria-label="anchor" href="#report-parameters"></a>
</h2>
<p>For reproducibility, the parameters used in the analysis and the
computational environment details are documented.</p>
<div class="section level3">
<h3 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<p>The <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> prints out all packages loaded at the
time of analysis, as well as their versions.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.12.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.12.0  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Paris</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] kableExtra_1.4.0            dplyr_1.1.4                </span></span>
<span><span class="co">##  [3] msigdbr_10.0.2              factoextra_1.0.7           </span></span>
<span><span class="co">##  [5] ggplot2_3.5.2               biomaRt_2.64.0             </span></span>
<span><span class="co">##  [7] CAIBIrnaseq_1.0.0           R.utils_2.13.0             </span></span>
<span><span class="co">##  [9] R.oo_1.27.1                 R.methodsS3_1.8.2          </span></span>
<span><span class="co">## [11] airway_1.28.0               SummarizedExperiment_1.38.1</span></span>
<span><span class="co">## [13] Biobase_2.68.0              GenomicRanges_1.60.0       </span></span>
<span><span class="co">## [15] GenomeInfoDb_1.44.0         IRanges_2.42.0             </span></span>
<span><span class="co">## [17] S4Vectors_0.46.0            BiocGenerics_0.54.0        </span></span>
<span><span class="co">## [19] generics_0.1.3              MatrixGenerics_1.20.0      </span></span>
<span><span class="co">## [21] matrixStats_1.5.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] ggplotify_0.1.2             filelock_1.0.3             </span></span>
<span><span class="co">##   [3] tibble_3.2.1                graph_1.86.0               </span></span>
<span><span class="co">##   [5] XML_3.99-0.18               lifecycle_1.0.4            </span></span>
<span><span class="co">##   [7] httr2_1.1.2                 rstatix_0.7.2              </span></span>
<span><span class="co">##   [9] lattice_0.22-5              crosstalk_1.2.1            </span></span>
<span><span class="co">##  [11] backports_1.5.0             magrittr_2.0.3             </span></span>
<span><span class="co">##  [13] plotly_4.10.4               sass_0.4.10                </span></span>
<span><span class="co">##  [15] rmarkdown_2.29              jquerylib_0.1.4            </span></span>
<span><span class="co">##  [17] yaml_2.3.10                 rlist_0.4.6.2              </span></span>
<span><span class="co">##  [19] cowplot_1.1.3               DBI_1.2.3                  </span></span>
<span><span class="co">##  [21] RColorBrewer_1.1-3          lubridate_1.9.4            </span></span>
<span><span class="co">##  [23] abind_1.4-8                 purrr_1.0.4                </span></span>
<span><span class="co">##  [25] yulab.utils_0.2.0           rappdirs_0.3.3             </span></span>
<span><span class="co">##  [27] GenomeInfoDbData_1.2.14     ggrepel_0.9.6              </span></span>
<span><span class="co">##  [29] irlba_2.3.5.1               tidytree_0.4.6             </span></span>
<span><span class="co">##  [31] GSVA_2.2.0                  MCPcounter_1.2.0           </span></span>
<span><span class="co">##  [33] annotate_1.86.0             svglite_2.1.3              </span></span>
<span><span class="co">##  [35] pkgdown_2.1.2               codetools_0.2-20           </span></span>
<span><span class="co">##  [37] DelayedArray_0.34.1         xml2_1.3.8                 </span></span>
<span><span class="co">##  [39] tidyselect_1.2.1            aplot_0.2.5                </span></span>
<span><span class="co">##  [41] UCSC.utils_1.4.0            farver_2.1.2               </span></span>
<span><span class="co">##  [43] ScaledMatrix_1.16.0         BiocFileCache_2.16.0       </span></span>
<span><span class="co">##  [45] jsonlite_2.0.0              Formula_1.2-5              </span></span>
<span><span class="co">##  [47] systemfonts_1.2.3           tools_4.5.0                </span></span>
<span><span class="co">##  [49] progress_1.2.3              treeio_1.32.0              </span></span>
<span><span class="co">##  [51] ragg_1.4.0                  Rcpp_1.0.14                </span></span>
<span><span class="co">##  [53] glue_1.8.0                  gridExtra_2.3              </span></span>
<span><span class="co">##  [55] SparseArray_1.8.0           xfun_0.52                  </span></span>
<span><span class="co">##  [57] DESeq2_1.48.0               HDF5Array_1.36.0           </span></span>
<span><span class="co">##  [59] withr_3.0.2                 fastmap_1.2.0              </span></span>
<span><span class="co">##  [61] rhdf5filters_1.20.0         digest_0.6.37              </span></span>
<span><span class="co">##  [63] rsvd_1.0.5                  timechange_0.3.0           </span></span>
<span><span class="co">##  [65] R6_2.6.1                    gridGraphics_0.5-1         </span></span>
<span><span class="co">##  [67] textshaping_1.0.1           RSQLite_2.3.10             </span></span>
<span><span class="co">##  [69] h5mread_1.0.0               tidyr_1.3.1                </span></span>
<span><span class="co">##  [71] data.table_1.17.0           prettyunits_1.2.0          </span></span>
<span><span class="co">##  [73] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [75] S4Arrays_1.8.0              pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [77] gtable_0.3.6                progeny_1.30.0             </span></span>
<span><span class="co">##  [79] blob_1.2.4                  SingleCellExperiment_1.30.0</span></span>
<span><span class="co">##  [81] XVector_0.48.0              htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [83] carData_3.0-5               fgsea_1.34.0               </span></span>
<span><span class="co">##  [85] msigdbdf_24.1.0             GSEABase_1.70.0            </span></span>
<span><span class="co">##  [87] scales_1.4.0                tidyverse_2.0.0            </span></span>
<span><span class="co">##  [89] png_0.1-8                   SpatialExperiment_1.18.0   </span></span>
<span><span class="co">##  [91] ggfun_0.1.8                 knitr_1.50                 </span></span>
<span><span class="co">##  [93] rstudioapi_0.17.1           tzdb_0.5.0                 </span></span>
<span><span class="co">##  [95] reshape2_1.4.4              rjson_0.2.23               </span></span>
<span><span class="co">##  [97] nlme_3.1-168                curl_6.2.2                 </span></span>
<span><span class="co">##  [99] cachem_1.1.0                rhdf5_2.52.0               </span></span>
<span><span class="co">## [101] stringr_1.5.1               parallel_4.5.0             </span></span>
<span><span class="co">## [103] vipor_0.4.7                 AnnotationDbi_1.70.0       </span></span>
<span><span class="co">## [105] desc_1.4.3                  pillar_1.10.2              </span></span>
<span><span class="co">## [107] grid_4.5.0                  vctrs_0.6.5                </span></span>
<span><span class="co">## [109] ggpubr_0.6.0                BiocSingular_1.24.0        </span></span>
<span><span class="co">## [111] car_3.1-3                   dbplyr_2.5.0               </span></span>
<span><span class="co">## [113] beachmat_2.24.0             xtable_1.8-4               </span></span>
<span><span class="co">## [115] beeswarm_0.4.0              evaluate_1.0.3             </span></span>
<span><span class="co">## [117] readr_2.1.5                 magick_2.8.6               </span></span>
<span><span class="co">## [119] cli_3.6.5                   locfit_1.5-9.12            </span></span>
<span><span class="co">## [121] compiler_4.5.0              rlang_1.1.6                </span></span>
<span><span class="co">## [123] crayon_1.5.3                ggsignif_0.6.4.9000        </span></span>
<span><span class="co">## [125] labeling_0.4.3              plyr_1.8.9                 </span></span>
<span><span class="co">## [127] forcats_1.0.0               fs_1.6.6                   </span></span>
<span><span class="co">## [129] ggbeeswarm_0.7.2            stringi_1.8.7              </span></span>
<span><span class="co">## [131] viridisLite_0.4.2           BiocParallel_1.42.0        </span></span>
<span><span class="co">## [133] assertthat_0.2.1            babelgene_22.9             </span></span>
<span><span class="co">## [135] Biostrings_2.76.0           lazyeval_0.2.2             </span></span>
<span><span class="co">## [137] ggheatmapper_0.2.1          Matrix_1.7-3               </span></span>
<span><span class="co">## [139] hms_1.1.3                   patchwork_1.3.0            </span></span>
<span><span class="co">## [141] sparseMatrixStats_1.20.0    bit64_4.6.0-1              </span></span>
<span><span class="co">## [143] Rhdf5lib_1.30.0             KEGGREST_1.48.0            </span></span>
<span><span class="co">## [145] broom_1.0.8                 memoise_2.0.1              </span></span>
<span><span class="co">## [147] bslib_0.9.0                 ggtree_3.16.0              </span></span>
<span><span class="co">## [149] fastmatch_1.1-6             bit_4.6.0                  </span></span>
<span><span class="co">## [151] ape_5.8-1</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Clarice Groeneveld, Chloe Witz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
