<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ CAIBIrnaseq</title><!-- mathjax math --><script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script><script>
  window.MathJax = {
    chtml: {
      fontURL: "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/output/chtml/fonts/woff-v2"
    }
  };
</script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet"><link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CAIBIrnaseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation Guide</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">References</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-examples" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-examples"><li><a class="dropdown-item" href="../articles/unsup_rnaseq.html">Unsupervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/sup_rnaseq.html">Supervised analysis</a></li>
    <li><a class="dropdown-item" href="../articles/venn.html">Comparing datasets</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="filter_gexp.html">filter_gexp()</a></code>

        </dt>
        <dd>Filter Gene Expression Data</dd>
      </dl><dl><dt>

          <code><a href="normalize_gexp.html">normalize_gexp()</a></code>

        </dt>
        <dd>Normalize Gene Expression Data</dd>
      </dl><dl><dt>

          <code><a href="gexp_preprocess.html">gexp_preprocess()</a></code>

        </dt>
        <dd>Pre-process a gene expression matrix to have different gene annotation</dd>
      </dl><dl><dt>

          <code><a href="rebase_gexp.html">rebase_gexp()</a></code>

        </dt>
        <dd>Rebase Gene Expression Matrix</dd>
      </dl><dl><dt>

          <code><a href="fixMarsExcel.html">fixMarsExcel()</a></code>

        </dt>
        <dd>Fix Excel-Corrupted Gene Names</dd>
      </dl><dl><dt>

          <code><a href="robust_cv.html">robust_cv()</a></code>

        </dt>
        <dd>Robust Coefficient of Variation</dd>
      </dl><dl><dt>

          <code><a href="highly_variable_genes.html">highly_variable_genes()</a></code>

        </dt>
        <dd>Identify Highly Variable Genes</dd>
      </dl><dl><dt>

          <code><a href="estimateTPM.html">estimateTPM()</a></code>

        </dt>
        <dd>Estimate Transcripts Per Million (TPM)</dd>
      </dl><dl><dt>

          <code><a href="readsPerMillion.html">readsPerMillion()</a></code>

        </dt>
        <dd>Normalize Counts to Reads Per Million (RPM)</dd>
      </dl><dl><dt>

          <code><a href="transcripts_per_million.html">transcripts_per_million()</a></code>

        </dt>
        <dd>Transcripts Per Million (TPM) Calculation</dd>
      </dl></div><div class="section level2">
      <h2 id="exploratory-analysis">Exploratory Analysis<a class="anchor" aria-label="anchor" href="#exploratory-analysis"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="pca_gexp.html">pca_gexp()</a></code>

        </dt>
        <dd>Perform PCA on Gene Expression Data</dd>
      </dl><dl><dt>

          <code><a href="plot_pca.html">plot_pca()</a></code>

        </dt>
        <dd>Plot PCA Results for Gene Expression Data</dd>
      </dl><dl><dt>

          <code><a href="cluster_exp.html">cluster_exp()</a></code>

        </dt>
        <dd>Cluster Samples Based on Gene Expression</dd>
      </dl><dl><dt>

          <code><a href="cluster_k_hc.html">cluster_k_hc()</a></code>

        </dt>
        <dd>Hierarchical Clustering with Optional PCA Dimensionality Reduction</dd>
      </dl><dl><dt>

          <code><a href="cluster_metadata.html">cluster_metadata()</a></code>

        </dt>
        <dd>Cluster Metadata in Expression Data</dd>
      </dl></div><div class="section level2">
      <h2 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="diffExpAnalysis.html">diffExpAnalysis()</a></code>

        </dt>
        <dd>Differential Expression Analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_exp_volcano.html">plot_exp_volcano()</a></code>

        </dt>
        <dd>Plot Volcano Plot of Differential Expression Results</dd>
      </dl><dl><dt>

          <code><a href="plot_exp_boxplot.html">plot_exp_boxplot()</a></code>

        </dt>
        <dd>Plot Expression Boxplot (with validation)</dd>
      </dl><dl><dt>

          <code><a href="plot_exp_heatmap.html">plot_exp_heatmap()</a></code>

        </dt>
        <dd>Plot Expression Heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_exp_scatter.html">plot_exp_scatter()</a></code>

        </dt>
        <dd>Plot Expression Scatter Plot</dd>
      </dl><dl><dt>

          <code><a href="plot_qc_filters.html">plot_qc_filters()</a></code>

        </dt>
        <dd>Plot QC Filters for Gene Expression Data</dd>
      </dl><dl><dt>

          <code><a href="plot_venn.html">plot_venn()</a></code>

        </dt>
        <dd>Plot a Venn diagram for two vectors</dd>
      </dl></div><div class="section level2">
      <h2 id="pathway-analysis">Pathway Analysis<a class="anchor" aria-label="anchor" href="#pathway-analysis"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="pathwayAnalysis.html">pathwayAnalysis()</a></code>

        </dt>
        <dd>Pathway Analysis</dd>
      </dl><dl><dt>

          <code><a href="pathwayFGSEA.html">pathwayFGSEA()</a></code>

        </dt>
        <dd>Pathway Enrichment Using FGSEA</dd>
      </dl><dl><dt>

          <code><a href="pathwayORA.html">pathwayORA()</a></code>

        </dt>
        <dd>Pathway Over-Representation Analysis (ORA)</dd>
      </dl><dl><dt>

          <code><a href="score_pathways.html">score_pathways()</a></code>

        </dt>
        <dd>Score Pathways</dd>
      </dl><dl><dt>

          <code><a href="score_progeny.html">score_progeny()</a></code>

        </dt>
        <dd>Score PROGENy Pathways</dd>
      </dl><dl><dt>

          <code><a href="get_pathway_df.html">get_pathway_df()</a></code>

        </dt>
        <dd>Extract pathway scores and metadata from a SummarizedExperiment</dd>
      </dl><dl><dt>

          <code><a href="get_pathway_genes.html">get_pathway_genes()</a></code>

        </dt>
        <dd>Get Pathway Genes</dd>
      </dl><dl><dt>

          <code><a href="plot_pathway_dotplot.html">plot_pathway_dotplot()</a></code>

        </dt>
        <dd>Plot Pathway Analysis Results as a Dot Plot</dd>
      </dl><dl><dt>

          <code><a href="plot_pathway_heatmap.html">plot_pathway_heatmap()</a></code>

        </dt>
        <dd>Plot Pathway Heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_path_boxplot.html">plot_path_boxplot()</a></code>

        </dt>
        <dd>Plot Pathway Boxplot</dd>
      </dl><dl><dt>

          <code><a href="plot_path_scatter.html">plot_path_scatter()</a></code>

        </dt>
        <dd>Plot Pathway Scatter Plot</dd>
      </dl></div><div class="section level2">
      <h2 id="microenvironment-cell-type-scores">Microenvironment &amp; Cell-type Scores<a class="anchor" aria-label="anchor" href="#microenvironment-cell-type-scores"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="mcp_counter.html">mcp_counter()</a></code>

        </dt>
        <dd>MCP-counter wrapper for SummarizedExperiment objects</dd>
      </dl><dl><dt>

          <code><a href="plot_microenv_heatmap.html">plot_microenv_heatmap()</a></code>

        </dt>
        <dd>Plot Microenvironment Heatmap</dd>
      </dl><dl><dt>

          <code><a href="plot_progeny_heatmap.html">plot_progeny_heatmap()</a></code>

        </dt>
        <dd>Plot Progeny Heatmap</dd>
      </dl></div><div class="section level2">
      <h2 id="utilities-and-helpers">Utilities and Helpers<a class="anchor" aria-label="anchor" href="#utilities-and-helpers"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="get_exp_df.html">get_exp_df()</a></code>

        </dt>
        <dd>Extract gene expression and sample annotations</dd>
      </dl><dl><dt>

          <code><a href="get_annotation_collection.html">get_annotation_collection()</a></code>

        </dt>
        <dd>Get Annotation Collection</dd>
      </dl><dl><dt>

          <code><a href="getEmptyRows.html">getEmptyRows()</a></code>

        </dt>
        <dd>Get Empty Rows in a Matrix</dd>
      </dl><dl><dt>

          <code><a href="plt_boxplot.html">plt_boxplot()</a></code>

        </dt>
        <dd>Boxplot of gene expression by annotation group</dd>
      </dl><dl><dt>

          <code><a href="plt_heatmap.html">plt_heatmap()</a></code>

        </dt>
        <dd>Plot Heatmap with Optional Annotations and Customization</dd>
      </dl><dl><dt>

          <code><a href="plt_scatter.html">plt_scatter()</a></code>

        </dt>
        <dd>Scatter plot of gene expression</dd>
      </dl><dl><dt>

          <code><a href="prep_exp_hm.html">prep_exp_hm()</a></code>

        </dt>
        <dd>Prepare Gene Expression Data for Heatmap Visualization</dd>
      </dl><dl><dt>

          <code><a href="prep_scoredf_hm.html">prep_scoredf_hm()</a></code>

        </dt>
        <dd>Prepare Pathway Scores Data Frame for Heatmap</dd>
      </dl><dl><dt>

          <code><a href="prep_scores_hm.html">prep_scores_hm()</a></code>

        </dt>
        <dd>Prepare Pathway Scores for Heatmap Visualization</dd>
      </dl><dl><dt>

          <code><a href="read_rnaseq_out.html">read_rnaseq_out()</a></code>

        </dt>
        <dd>Read RNA-seq Output from nf-core/rnaseq</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Clarice Groeneveld, Chloe Witz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer></div>





  </body></html>

